<!DOCTYPE html>
<html lang="en">

<head>
  <title>
  Integrating Tree Data: Methods and Applications · YuLab@SMU
</title>
  <meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="color-scheme" content="light dark">




<meta name="author" content="YuLab-SMU">
<meta name="description" content="We have proposed and developed a series of methods and software tools for the operation, integration, and visualization of phylogenetic trees and data. Key innovations include: (1) introducing graphic grammar to the field of phylogenetics for the first time; (2) enhancing the data integration capabilities of phylogenetics and its application across various disciplines; (3) proposing two universal methods for phylogenetic data integration and visualization; and (4) designing data structures that can store phylogenetic trees, associated data, and visualization directives to ensure analytical reproducibility.">
<meta name="keywords" content="developer,bioinformatics,omics,visualization">


  <meta name="twitter:card" content="summary">
  <meta name="twitter:title" content="Integrating Tree Data: Methods and Applications">
  <meta name="twitter:description" content="We have proposed and developed a series of methods and software tools for the operation, integration, and visualization of phylogenetic trees and data. Key innovations include: (1) introducing graphic grammar to the field of phylogenetics for the first time; (2) enhancing the data integration capabilities of phylogenetics and its application across various disciplines; (3) proposing two universal methods for phylogenetic data integration and visualization; and (4) designing data structures that can store phylogenetic trees, associated data, and visualization directives to ensure analytical reproducibility.">

<meta property="og:url" content="http://yulab-smu.top/contribution-tree-data/">
  <meta property="og:site_name" content="YuLab@SMU">
  <meta property="og:title" content="Integrating Tree Data: Methods and Applications">
  <meta property="og:description" content="We have proposed and developed a series of methods and software tools for the operation, integration, and visualization of phylogenetic trees and data. Key innovations include: (1) introducing graphic grammar to the field of phylogenetics for the first time; (2) enhancing the data integration capabilities of phylogenetics and its application across various disciplines; (3) proposing two universal methods for phylogenetic data integration and visualization; and (4) designing data structures that can store phylogenetic trees, associated data, and visualization directives to ensure analytical reproducibility.">
  <meta property="og:locale" content="en">
  <meta property="og:type" content="article">




<link rel="canonical" href="http://yulab-smu.top/contribution-tree-data/">


<link rel="preload" href="/fonts/fa-brands-400.woff2" as="font" type="font/woff2" crossorigin>
<link rel="preload" href="/fonts/fa-regular-400.woff2" as="font" type="font/woff2" crossorigin>
<link rel="preload" href="/fonts/fa-solid-900.woff2" as="font" type="font/woff2" crossorigin>


  
  
  <link rel="stylesheet" href="/css/coder.min.38c4552ac40f9ae3408bad40358f654ebd8804412fe74ed56f2d6c8a7af82dd3.css" integrity="sha256-OMRVKsQPmuNAi61ANY9lTr2IBEEv507Vby1sinr4LdM=" crossorigin="anonymous" media="screen" />








 




<link rel="icon" type="image/svg+xml" href="/images/favicon.svg" sizes="any">
<link rel="icon" type="image/png" href="/images/favicon-32x32.png" sizes="32x32">
<link rel="icon" type="image/png" href="/images/favicon-16x16.png" sizes="16x16">

<link rel="apple-touch-icon" href="/images/apple-touch-icon.png">
<link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon.png">

<link rel="manifest" href="/site.webmanifest">
<link rel="mask-icon" href="/images/safari-pinned-tab.svg" color="#5bbad5">









</head>




<body class="preload-transitions colorscheme-light">
  
<div class="float-container">
    <a id="dark-mode-toggle" class="colorscheme-toggle">
        <i class="fa-solid fa-adjust fa-fw" aria-hidden="true"></i>
    </a>
</div>


  <main class="wrapper">
    <nav class="navigation">
  <section class="container">
    
    <a class="navigation-title" href="http://yulab-smu.top/">
      YuLab@SMU
    </a>
    
    
      <input type="checkbox" id="menu-toggle" />
      <label class="menu-button float-right" for="menu-toggle">
        <i class="fa-solid fa-bars fa-fw" aria-hidden="true"></i>
      </label>
      <ul class="navigation-list">
        
          
            <li class="navigation-item">
              <a class="navigation-link " href="/about/">About</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link " href="/publications/">Publications</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link " href="/contributions/">Contributions</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link " href="/members/">Members</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link " href="/news/">News</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link " href="/contact/">Contact</a>
            </li>
          
        
        
      </ul>
    
  </section>
</nav>


    <div class="content">
      
  <section class="container page">
  <article>
    <header>
      <h1 class="title">
        <a class="title-link" href="http://yulab-smu.top/contribution-tree-data/">
          Integrating Tree Data: Methods and Applications
        </a>
      </h1>
    </header>

    <style>
    @media only screen and (max-width: 768px) {
        td {
            display: block;
        }
    }
</style>
<p>We have proposed and developed a series of methods and software tools for the operation, integration, and visualization of phylogenetic trees and data. Key innovations include: (1) introducing graphic grammar to the field of phylogenetics for the first time; (2) enhancing the data integration capabilities of phylogenetics and its application across various disciplines; (3) proposing two universal methods for phylogenetic data integration and visualization; and (4) designing data structures that can store phylogenetic trees, associated data, and visualization directives to ensure analytical reproducibility. These methods and tools offer a concise and unified syntax system, assisting researchers in discovering hidden patterns and proposing new hypotheses by integrating heterogeneous data within the context of evolution or hierarchy.</p>
<p>Two monographs have been published to introduce this series of work: &ldquo;<a href="https://www.routledge.com/Data-Integration-Manipulation-and-Visualization-of-Phylogenetic-Trees/Yu/p/book/9781032233574"  class="external-link" target="_blank" rel="noopener">Data integration, manipulation and visualization of phylogenetic trees</a>&rdquo; (in English) by CRC Press and 《<a href="https://weread.qq.com/web/bookDetail/8ad32a00813ab81bbg0183d2"  class="external-link" target="_blank" rel="noopener">R实战：系统发育树的数据集成操作与可视化</a>》 (in Chinese) by Publishing House of Electronics Industry (电子工业出版社).</p>
<p><a href="https://www.routledge.com/Data-Integration-Manipulation-and-Visualization-of-Phylogenetic-Trees/Yu/p/book/9781032233574"><img src="https://yulab-smu.top/treedata-book/9781032233574_cover_review.png" style="height:320px;border:2px solid black;"/></a>
<a href="https://weread.qq.com/web/bookDetail/8ad32a00813ab81bbg0183d2"><img src="/images/book-cover-cn.png" style="height:320px;border:2px solid black;"/></a></p>
<hr>
<h2 id="1-frist-time-implementation-of-graphical-grammar-for-visualizing-phylogenetic-tree-data">
  1. Frist-time implementation of graphical grammar for visualizing phylogenetic tree data
  <a class="heading-link" href="#1-frist-time-implementation-of-graphical-grammar-for-visualizing-phylogenetic-tree-data">
    <i class="fa-solid fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:25%">
<p><a href="http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/abstract"><img src="/images/ggtree/ggtree-2017.png" width='1000px'/></a></p>
</td>
<td style="border:none;">
<p>Numerous software tools exist for visualizing phylogenetic trees, but they primarily focus on displaying the tree&rsquo;s topological structure and often lack comprehensive support for annotating the tree with additional knowledge or data. For the first time, <a href="https://www.bioconductor.org/packages/ggtree"  class="external-link" target="_blank" rel="noopener">ggtree</a> introduces the grammar of graphics into the visualization of phylogenetic trees and related data. This innovative approach enables a high level of abstraction for visualization through a simple grammar, significantly reducing the complexity of data visualization and accommodating complex requirements. This work was published in <a href="http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/abstract"  class="external-link" target="_blank" rel="noopener"><em>Methods in Ecology and Evolution</em></a> in 2017 (ESI highly cited) and was chosen by the journal as <a href="https://methodsblog.com/2020/11/19/ggtree-tree-visualization/"  class="external-link" target="_blank" rel="noopener">one of the ten representative works</a> for its 10th anniversary celebration. An invited protocol paper demonstrating the use of this package was also published in <a href="https://currentprotocols.onlinelibrary.wiley.com/doi/abs/10.1002/cpbi.96"  class="external-link" target="_blank" rel="noopener"><em>Current Protocols in Bioinformatics</em></a> in 2020.</p>
</td>
</tr>
</table>
<!--
系统发育树可视化有非常多的软件实现，然而主要用于可视化树的拓扑结构，无法（或仅有限支持）利用知识和数据对树进行注释，ggtree首次将图形语法引入到系统发育树及相关数据的可视化，有效地实现了简单语法对可视化需求的高度抽象，大大降低了数据可视化的难度，让复杂的需求成为可能.该工作发表于 Methods in Ecology and Evolution 2017, 该论文被期刊选为 10 周年纪念 10 篇代表作之一。
-->
<h2 id="2-base-classes-and-functions-for-phylogenetic-tree-input-and-output">
  2. Base classes and functions for phylogenetic tree input and output
  <a class="heading-link" href="#2-base-classes-and-functions-for-phylogenetic-tree-input-and-output">
    <i class="fa-solid fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:25%">
<p><a href="https://academic.oup.com/mbe/article/37/2/599/5601621"><img src="/images/ggtree/treeio-2020.png" width='1000px'/></a></p>
</td>
<td style="border:none;">
<p>The outputs of phylogenetic software are often in non-standard formats, leading to compatibility issues and hindering integration and comparative analysis in downstream applications. To address this challenge, we developed <a href="https://www.bioconductor.org/packages/treeio"  class="external-link" target="_blank" rel="noopener">treeio</a>, a tool capable of parsing both standard and a variety of non-standard data formats. It facilitates the integration of external data and supports exporting phylogenetic trees and associated data into a single file. This dual capability enables data format conversion, thereby indirectly expanding the compatibility of software with a wider range of data. By parsing and integrating diverse data types, <a href="https://www.bioconductor.org/packages/treeio"  class="external-link" target="_blank" rel="noopener">treeio</a> empowers downstream integrated and comparative analyses, thus broadening the application scope of phylogenetic analysis. Our work, published in <a href="https://academic.oup.com/mbe/article/37/2/599/5601621"  class="external-link" target="_blank" rel="noopener"><em>Molecular Biology and Evolution</em></a> in 2020 (ESI highly cited), underscores the significance of this advancement.</p>
</td>
</tr>
</table>
<!--
系统发育相关的软件输出多数为非标准格式，互相之间无法解析，使得分析结果无法在不同的软件中使用，极大地限制了下游的整合与比较分析。针对这一问题，我们开发了基础的输入输出软件工具 － treeio，允许解析标准的数据格式以及十几种非标准的软件输出，并且允许整合外部数据。此外还支持将系统发育树与相关数据输出到单一的文件中。输入和输出的支持，意味着允许数据进行格式转换，进而让更多的软件变相地支持更多的数据。而多种数据的解析和整合，则为下游的整合分析、比较分析提供了可能性，也扩展了系统发育分析的应用范畴。将该工作发表于 Molecular Biology and Evolution 2020。
-->
<h2 id="3-proposing-two-general-methods-for-the-integration-and-visualization-of-phylogenetic-data">
  3. Proposing two general methods for the integration and visualization of phylogenetic data
  <a class="heading-link" href="#3-proposing-two-general-methods-for-the-integration-and-visualization-of-phylogenetic-data">
    <i class="fa-solid fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:25%">
<p><a href="https://doi.org/10.1093/molbev/msab166"><img src="/images/ggtree/ggtreeExtra-2021.png" width='1000px'/></a></p>
</td>
<td style="border:none;">
<p>Two comprehensive methods have been proposed and implemented to address all facets of integrating and visualizing phylogenetic data. The first method enables data to be directly mapped onto the tree&rsquo;s topology, facilitating the direct display of data or its transformation into visualization features. The second method involves the restructuring external data based on the tree&rsquo;s topology, allowing users to visualize it according to their specifications, and subsequently aligning the visualization with the phylogenetic tree. These two methods enable the integration of diverse heterogeneous data from various disciplines within the context of phylogenetics, thereby aiding in the discovery of new patterns or the formulation of novel hypotheses. This work, published in <a href="https://academic.oup.com/mbe/article/35/12/3041/5142656"  class="external-link" target="_blank" rel="noopener"><em>Molecular Biology and Evolution</em></a> in 2018 (ESI highly cited), served as the foundation for the development of the <a href="https://www.bioconductor.org/packages/ggtreeExtra"  class="external-link" target="_blank" rel="noopener">ggtreeExtra</a> package. This package enhances integration and visualization capabilities for data richly annotated with additional features, and was published in <a href="https://doi.org/10.1093/molbev/msab166"  class="external-link" target="_blank" rel="noopener"><em>Molecular Biology and Evolution</em></a> in 2021.</p>
</td>
</tr>
</table>
<!-- 提出并实现了两种通用的方法，涵盖了系统发育数据整合与可视化的方方面面。方法一允许将数据映射到树的拓扑结构上，并支持将数据直接展示或映射为可视化特征（如颜色、大小、粗细等）；方法二则将外部数据按照树的拓扑结构进行重排，再按照用户指定的方式进行可视化，最终将可视化结果与系统发育树对齐展示，方便研究人员结合系统发育信息对数据进行解读。两种通用的方法使得来自于不同学科的各种异质性的数据能够在系统发育的背景下得以解析，有助于发现与演化相关的新模式或提出新的假说。将该工作发表于Molecular Biology and Evolution 2018。在此基础上，又实现了ggtreeExtra包，以加强高维度数据的整合可视化能力，这一工作发表于Molecular Biology and Evolution 2021。
--> 
<h2 id="4-enhancing-data-reuse-and-analytical-reproducibility">
  4. Enhancing data reuse and analytical reproducibility
  <a class="heading-link" href="#4-enhancing-data-reuse-and-analytical-reproducibility">
    <i class="fa-solid fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:25%">
<p><a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/imt2.56"><img src="/images/ggtree/imeta-2022.png" width='1000px'/></a></p>
</td>
<td style="border:none;">
<p>Visualisation of phylogenetic trees typically manifests as static images, leading to a lack of reusability of both the trees and associated data, thus impeding the integration of phylogenetic knowledge and comparative analysis. Research indicates that approximately 60% of published phylogenetic data is lost permanently. To tackle this issue, we devised the <em>ggtree</em> object, which encapsulates the phylogenetic tree, data, and visualization directives. This object can be rendered into an image, from which the phylogenetic tree and related data can be extracted. Furthermore, akin to the &ldquo;format painter&rdquo; tool, the visualization directives are transferable for visualizing other tree objects. Our efforts,  published in <a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/imt2.56"  class="external-link" target="_blank" rel="noopener"><em>iMeta</em></a> in 2022, promote data reusability and research replicability, while facilitating the integration and comparative analysis of phylogenetic data within the field.</p>
</td>
</tr>
</table>
<!--

系统发育树的可视化通常以图片的形式，而相应的树和数据无法被重新使用，以整合系统发育学知识以及进行比较分析，有研究表明，约 60%的已发表系统发育数据永久地丢失了。为了解决这一问题，我们设计了 ggtree 对象，包含了系统发育树、数据和可视化指令，它可以渲染成一张图片，同时能够从中抽提出系统发育树和相关的数据，并且类似于“格式刷”，可视化指令还能用于可视化其它的树对象。这一工作发表于iMeta 2022，有力地支持数据的可重用性和研究的可重复性，同时也能够促进学科对系统发育数据的整合与比较分析。
-->
<h2 id="5-expanding-support-to-other-tree-like-structures">
  5. Expanding support to other tree-like structures
  <a class="heading-link" href="#5-expanding-support-to-other-tree-like-structures">
    <i class="fa-solid fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:25%">
<p><a href="https://www.bioconductor.org/packages/ggtreeDendro"><img src="/images/ggtree/ggtreeDendro.png" width='1000px'/></a></p>
</td>
<td style="border:none;">
<p>Broaden the scope of tools pertaining to tree data integration and visualization to encompass other tree-like structures, such as hierarchical clustering and classification/regression trees. We have developed the <a href="https://www.bioconductor.org/packages/ggtreeDendro"  class="external-link" target="_blank" rel="noopener">ggtreeDendro</a> package to accommodate general hierarchical structures and are currently working on the <a href="https://github.com/YuLab-SMU/ecluster"  class="external-link" target="_blank" rel="noopener">ecluster</a> package to support various omics data structures available in the Bioconductor project. These advancements enable the interpretation and integration of related data at the feature or sample level based on their hierarchical relationships.</p>
</td>
</tr>
</table>
<!--

扩展树相关数据集成与可视化这一系列工具以应用于其它tree-like structures (e.g., hierarchical clustering and classification/regression trees) ，实现了ggtreeDendro用于支持通用的层次结构和ecluster包用于Bioconductor各种组学数据结构的支持。这使得feature或样本水平的相关数据能够基于层次结构进行解析和整合。
-->
<hr>
<h2 id="feedback-from-the-academic-community">
  Feedback from the academic community
  <a class="heading-link" href="#feedback-from-the-academic-community">
    <i class="fa-solid fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:75%">
<blockquote>
<p>It is embarrassing to me that I just learned about &ldquo;<strong>ggtree</strong>&rdquo; this month, but I am looking forward to incorporating it into several projects in my lab.  I am looking forward to using it to visualizing annotation results after doing BLAST or FASTA similarity searches, and for high lighting errors in protein sequences among closely related organisms.  I have been searching for a tool like &ldquo;<strong>ggtree</strong>&rdquo; for several years, not appreciating that you had already built it.  So my interest in recruiting your paper is somewhat selfish.  I would like more help in understanding how to add graphical annotations to the leaves of trees.</p>
<p>&ndash; The email from Professor <a href="https://fasta.bioch.virginia.edu/wrpearson/"  class="external-link" target="_blank" rel="noopener">William Pearson</a> (ISCB Fellow), dated September 14, 2019, invited <a href="https://twitter.com/guangchuangyu"  class="external-link" target="_blank" rel="noopener">Guangchuang Yu</a> to contribute to <em>Current Protocols in Bioinformatics</em>.</p>
</blockquote>
</td>
<td style="border:none;">
<a href="https://fasta.bioch.virginia.edu/wrpearson/"><img src="https://fasta.bioch.virginia.edu/wrpearson/wrp_crop_2018.jpg" width='200px'/></a>
</td>
</tr>
</table>
<h2 id="publications">
  Publications
  <a class="heading-link" href="#publications">
    <i class="fa-solid fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<font size="3">
<ul>
<li>L Zhan<sup>#</sup>, X Luo<sup>#</sup>, W Xie<sup>#</sup>, XA Zhu<sup>#</sup>, Z Xie, J Lin, L Li, W Tang, R Wang, L Deng, Y Liao, B Liu, Y Cai, Q Wang, S Xu<sup>*</sup>, <strong>G Yu</strong><sup>*</sup>. <a href="https://www.sciencedirect.com/science/article/pii/S167385272400033X"  class="external-link" target="_blank" rel="noopener">shinyTempSignal: an R shiny application for exploring temporal and other phylogenetic signals</a>. <strong><em>Journal of Genetics and Genomics</em></strong>. 2024, 51(7):762-768.</li>
<li>L Li, W Xie, L Zhan, S Wen, X Luo, S Xu, Y Cai, W Tang, Q Wang, M Li, Z Xie, L Deng, H Zhu, <strong>G Yu</strong><sup>*</sup>. <a href="https://www.sciencedirect.com/science/article/pii/S2667005424000206"  class="external-link" target="_blank" rel="noopener">Resolving Tumor Evolution: A Phylogenetic Approach</a>. <strong><em>Journal of the National Cancer Center</em></strong>. 2024, 4(2):97-106.</li>
<li>S Xu, L Li, X Luo, M Chen, W Tang, L Zhan, Z Dai, TT. Lam, Y Guan, <strong>G Yu</strong><sup>*</sup>. <a href="https://onlinelibrary.wiley.com/doi/full/10.1002/imt2.56"  class="external-link" target="_blank" rel="noopener">Ggtree: A serialized data object for visualization of a phylogenetic tree and annotation data</a>. <strong><em>iMeta</em></strong>, 2022, 1(4):e56.</li>
<li><strong>G Yu</strong>. Data Integration, Manipulation and Visualization of Phylogenetic Treess (1st edition). <strong><em>Chapman and Hall/CRC</em></strong>, 2022. doi: <a href="https://doi.org/10.1201/9781003279242"  class="external-link" target="_blank" rel="noopener">10.1201/9781003279242</a></li>
<li>L Zhou<sup>#</sup>, T Feng<sup>#</sup>, S Xu, F Gao, TT Lam, Q Wang, T Wu, H Huang, L Zhan, L Li, Y Guan, Z Dai<sup>*</sup>, <strong>G Yu</strong><sup>*</sup>. <a href="https://doi.org/10.1093/bib/bbac222"  class="external-link" target="_blank" rel="noopener">ggmsa: a visual exploration tool for multiple sequence alignment and associated data</a>. <strong><em>Briefings in Bioinformatics</em></strong>. 2022, 23(4):bbac222.</li>
<li>S Xu, Z Dai, P Guo, X Fu, S Liu, L Zhou, W Tang, T Feng, M Chen, L Zhan, T Wu, E Hu, Y Jiang<sup>*</sup>, X Bo<sup>*</sup>, <strong>G Yu</strong><sup>*</sup>. <a href="https://doi.org/10.1093/molbev/msab166"  class="external-link" target="_blank" rel="noopener">ggtreeExtra: Compact visualization of richly annotated phylogenetic data</a>. <strong><em>Molecular Biology and Evolution</em></strong>. 2021, 38(9):4039-4042.</li>
<li><strong>G Yu</strong><sup>*</sup>. <a href="https://doi.org/10.1002/cpbi.96"  class="external-link" target="_blank" rel="noopener">Using ggtree to Visualize Data on Tree-Like Structure</a>. <strong><em>Current Protocols in Bioinformatics</em></strong>. 2020, 69(1):e96.</li>
<li>LG Wang, TTY Lam, S Xu, Z Dai, L Zhou, T Feng, P Guo, CW Dunn, BR Jones, T Bradley, H Zhu, Y Guan, Y Jiang, <strong>G Yu</strong><sup>*</sup>. <a href="https://doi.org/10.1093/molbev/msz240"  class="external-link" target="_blank" rel="noopener">treeio: an R package for phylogenetic tree input and output with richly annotated and associated data</a>. <strong><em>Molecular Biology and Evolution</em></strong>. 2020, 37(2):599-603.</li>
<li><strong>G Yu</strong><sup>*</sup>, TTY Lam, H Zhu, Y Guan<sup>*</sup>. <a href="https://academic.oup.com/mbe/article-abstract/35/12/3041/5142656"  class="external-link" target="_blank" rel="noopener">Two methods for mapping and visualizing associated data on phylogeny using ggtree</a>. <strong><em>Molecular Biology and Evolution</em></strong>. 2018, 35(12):3041-3043.</li>
<li><strong>G Yu</strong>, DK Smith, H Zhu, Y Guan, TTY Lam<sup>*</sup>. <a href="http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/abstract"  class="external-link" target="_blank" rel="noopener">ggtree: an R package for visualization and annotation of phylogenetic trees with their covariates and other associated data</a>. <strong><em>Methods in Ecology and Evolution</em></strong>. 2017, 8(1):28-36.</li>
</ul>
</font>

  </article>
</section>

  

    </div>

    <footer class="footer">
  <section class="container">
    ©
    
      2018 -
    
    2024
     YuLab-SMU 
    ·
    
    Powered by <a href="https://gohugo.io/" target="_blank" rel="noopener">Hugo</a> & <a href="https://github.com/luizdepra/hugo-coder/" target="_blank" rel="noopener">Coder</a>.
    
  </section>
</footer>

  </main>

  

  
  
  <script src="/js/coder.min.6ae284be93d2d19dad1f02b0039508d9aab3180a12a06dcc71b0b0ef7825a317.js" integrity="sha256-auKEvpPS0Z2tHwKwA5UI2aqzGAoSoG3McbCw73gloxc="></script>
  

  

  


  
  



  

  

  <script data-goatcounter="https://code.goatcounter.com/count"
        async src="//gc.zgo.at/count.js"></script>


  
<script defer src='https://static.cloudflareinsights.com/beacon.min.js'
        data-cf-beacon='{"token": "token"}'></script>



  

  

  

  

  

  

  

  

  

  

  

<a href="https://clustrmaps.com/site/1c1ic"  title="Visit tracker"><img src="//www.clustrmaps.com/map_v2.png?d=-8uXb2irPM0FIvaNbU25vKNLCgWjB60Wf8-JCJVmzUc&cl=ffffff" /></a>
</body>

</html>
