---
slug: publications
title: Publications
---


<style>
    @media only screen and (max-width: 768px) {
        td {
            display: block;
        }
    }
</style>


<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;">

+ <sup>#</sup> Co-first authors
+ <sup>\*</sup> Co-corresponding authors 
+ <i class="fa fa-trophy fa-2x" aria-hidden="true" style="color:#CC5500;font-size: 150%;"></i>：ESI Highly Cited Paper (9X)
+ 分区：文章发表当年的中科院小类分区
+ IF：文章发表当年的影响因子
+ [课题组学术榜](/awards)

</td>
<td style="border:none;">
<a href="https://scholar.google.com/citations?user=DO5oG40AAAAJ"><img src="/images/citation.png" height='200px'/></a>
</td>
</tr>
</table>


## <i class="fa fa-calendar fa-1x" aria-hidden="true"></i> 2025 {#year-2025}

+ D Zeng<sup>#</sup>, Y Fang<sup>#</sup>, W Qiu; P Luo, S Wang; R Shen, W Gu, X Huang, Q Mao; G Wang, Y Lai; X Xu, M Shi, Z Wu<sup>\*</sup>, __G Yu__<sup>\*</sup>, W Liao<sup>\*</sup>. [Enhancing Immuno-Oncology Investigations Through Multidimensional Decoding of Tumour Microenvironment with IOBR 2.0](). __*Cell Reports Methods*__. accepted.

## <i class="fa fa-calendar fa-1x" aria-hidden="true"></i> 2024 {#year-2024}

+ __G Yu__<sup>\*</sup>. [Thirteen years of clusterProfiler](https://doi.org/10.1016/j.xinn.2024.100722). __*The Innovation*__. 2024, 5(6):100722.
+ Z Miao<sup>#</sup>, T Tian<sup>#</sup>, W Chen<sup>#</sup>, Q Wang<sup>#</sup>, L Ma<sup>#</sup>, D Zhang<sup>#</sup>, M Xie<sup>#</sup>, Z Yu<sup>#</sup>, X Guo, G Bai, S Zhao, X Chen, W Wang, Y Gao, S Guo<sup>\*</sup>, M Luo<sup>\*</sup>, L Yuan<sup>\*</sup>, C Tian<sup>\*</sup>, L Wu<sup>\*</sup>, __G Yu__<sup>\*</sup>, D Zhang<sup>\*</sup>, S Sun<sup>\*</sup>. [Spatial Resolved Transcriptomics: Computational Insights into Gene Transcription Across Tissue and Organ Architecture in Diverse Applications](https://www.the-innovation.org/article/doi/10.59717/j.xinn-life.2024.100097). __*The Innovation Life*__. 2024, 2(4):100097.
+ S Xu<sup>#</sup>, E Hu<sup>#</sup>, Y Cai<sup>#</sup>, Z Xie<sup>#</sup>, X Luo<sup>#</sup>, L Zhan, W Tang, Q Wang, B Liu, R Wang, W Xie, T Wu, L Xie, __G Yu__<sup>\*</sup>. [Using clusterProfiler to characterise Multi-Omics Data](https://www.nature.com/articles/s41596-024-01020-z). __*Nature Protocols*__. 2024, 19(11):3292-3320. 
+ Y Chen<sup>#</sup>, LN He<sup>#</sup>, Y Zhang, J Gong, S Xu, Y Shu, D Zhang<sup>\*</sup>, __G Yu__<sup>\*</sup>, Z Zuo<sup>\*</sup>. [tigeR: Tumor Immunotherapy Gene Expression Data Analysis R package](http://doi.org/10.1002/imt2.229). __*iMeta*__. 2024, 3(5):e229.
+ D Wang, G Chen, L Li, S Wen, Z Xie, X Luo, L Zhan, S Xu, J Li, R Wang, Q Wang<sup>\*</sup>, __G Yu__<sup>\*</sup>. [Reducing language barriers, promoting information absorption, and communication using *fanyi*](https://journals.lww.com/cmj/fulltext/2024/08200/reducing_language_barriers,_promoting_information.7.aspx). __*Chinese Medical Journal*__. 2024, 137(16):1950-1956.
+ XK Ma<sup>#</sup>, Y Yu<sup>#</sup>, T Huang<sup>#</sup><sup>\*</sup>, D Zhang, C Tian, W Tang, M Luo, P Du<sup>\*</sup>, __G Yu__<sup>\*</sup>, L Yang<sup>\*</sup>. [Bioinformatics Software Development: principles and future directions](https://www.the-innovation.org/article/doi/10.59717/j.xinn-life.2024.100083). __*The Innovation Life*__. 2024, 2(3):100083.
+ Y Yang<sup>#</sup>, S Xu<sup>#</sup>, Y Hong, Y Cai, W Tang, J Wang, B Shen<sup>\*</sup>, H Zong<sup>\*</sup>, __G Yu__<sup>\*</sup>. [Computational modeling for medical data: from data collection to knowledge discovery](https://doi.org/10.59717/j.xinn-life.2024.100079). __*The Innovation Life*__. 2024, 2(3):100079.
+ L Zhan<sup>#</sup>, X Luo<sup>#</sup>, W Xie<sup>#</sup>, XA Zhu<sup>#</sup>, Z Xie, J Lin, L Li, W Tang, R Wang, L Deng, Y Liao, B Liu, Y Cai, Q Wang, S Xu<sup>\*</sup>, __G Yu__<sup>\*</sup>. [shinyTempSignal: an R shiny application for exploring temporal and other phylogenetic signals](https://www.sciencedirect.com/science/article/pii/S167385272400033X). __*Journal of Genetics and Genomics*__. 2024, 51(7):762-768.
+ L Li, W Xie, L Zhan, S Wen, X Luo, S Xu, Y Cai, W Tang, Q Wang, M Li, Z Xie, L Deng, H Zhu, __G Yu__<sup>\*</sup>. [Resolving Tumor Evolution: A Phylogenetic Approach](https://www.sciencedirect.com/science/article/pii/S2667005424000206). __*Journal of the National Cancer Center*__. 2024, 4(2):97-106. 
+ Q Wang, W Tang, L Chen, L Deng, __G Yu__<sup>\*</sup>. [Decoding the Brain: Unveiling Comprehensive Cellular Atlases through Multiomics Mega Data](https://doi.org/10.1016/j.xinn.2024.100637). __*The Innovation*__. 2024, 5(4):100637.
+ Y Cai<sup>#</sup>, W Hu<sup>#</sup>, Y Pei, Hao Zhao<sup>\*</sup>, __G Yu__<sup>\*</sup>. [Encoding biological metaverse: Advancements and challenges in neural fields from macroscopic to microscopic](https://doi.org/10.1016/j.xinn.2024.100627). __*The Innovation*__. 2024, 5(3):100627.
+ Y Lai<sup>#</sup>, W Tang<sup>#</sup>, X Luo, H Zheng, Y Zhang<sup>\*</sup>, M Wang<sup>\*</sup>, __G Yu__<sup>\*</sup>, Min Yang<sup>\*</sup>. [Gut microbiome and metabolome to discover pathogenic bacteria and probiotics in ankylosing spondylitis](https://www.frontiersin.org/journals/immunology/articles/10.3389/fimmu.2024.1369116). __*Frontiers in Immunology*__. 2024, 15:1369116.
+ CH Gao, C Chen, T Akyol, A Dusa, __G Yu__, B Cao, P Cai<sup>\*</sup>. [ggVennDiagram: intuitive Venn diagram software extended](https://onlinelibrary.wiley.com/doi/10.1002/imt2.177). __*iMeta*__. 2024, 3(1):e177.
+ Y Ye<sup>#</sup>, MH Shum<sup>#</sup>, JL Tsui<sup>#</sup>, __G Yu__<sup>#</sup>, DK Smith, H Zhu, JT Wu, Y Guan, TTY Lam<sup>\*</sup>. [Robust expansion of phylogeny for fast-growing genome sequence data](https://doi.org/10.1371/journal.pcbi.1011871). __*PLoS Computational Biology*__. 2024, 20(2):e1011871.
+ S Wen, S Mo, J Zhou, Y Lv, K Khazaie<sup>\*</sup>, __G Yu__<sup>\*</sup>. [Single-cell and spatial-omics in delineating immune-related diseases](https://www.frontiersin.org/articles/10.3389/fcell.2024.1365242/full). __*Frontiers in Cell and Developmental Biology*__. 2024, 12:1365242.


## <i class="fa fa-calendar fa-1x" aria-hidden="true"></i> 2023 {#year-2023}

+ X Zhang, B Liang, Y Huang, H Meng, Z Li, J Du, L Zhou, Y Zhong, B Wang, X Lin, __G Yu__, X Chen, W Lu, ZS Chen, X Yang, Z Huang<sup>\*</sup>. [Behind the Indolent Facade: Uncovering the Molecular Features and Malignancy Potential in Lung Minimally Invasive Adenocarcinoma by Single-Cell Transcriptomics](https://doi.org/10.1002/advs.202303753). __*Advanced Science*__. 2023, 10(36):2303753. (IF = 14.3，一区)
+ M Kang, F Lin, Z Jiang, X Tan, X Lin, Z Liang, C Xiao, Y Xia, W Guan, Z Yang, __G Yu__, M Zanin, S Tang<sup>\*</sup>, SS Wong<sup>\*</sup>. [The impact of pre-existing influenza antibodies and inflammatory status on the influenza vaccine responses in older adults](https://onlinelibrary.wiley.com/doi/abs/10.1111/irv.13172). __*Influenza and Other Respiratory Viruses*__. 2023, 17(7):e13172. (IF = 4.3，四区)
+ W Tang, H Zheng, S Xu, P Li, L Zhan, X Luo, Z Dai, Q Wang, __G Yu__<sup>\*</sup>. [MMINP: A computational framework of microbe-Metabolite interactions-based metabolic profiles predictor based on the O2-PLS algorithm](https://dx.doi.org/10.1080/19490976.2023.2223349). __*Gut Microbes*__. 2023, 15(1):2223349. (IF = 12.2，一区)
+ S Xu, L Zhan, W Tang, Q Wang, Z Dai, L Zhou, T Feng, M Chen, T Wu, E Hu, __G Yu__<sup>\*</sup>. [MicrobiotaProcess: A comprehensive R package for deep mining microbiome](https://doi.org/10.1016/j.xinn.2023.100388). __*The Innovation*__. 2023, 4(2):100388. (IF = 33.2，一区)
+ S Guo<sup>\*</sup>, D Zhang, H Wang, Q An, __G Yu__, J Han, C Jiang, J Huang. [Computational and Systematic Analysis of Multi-omics Data for Drug Discovery and Development](https://www.frontiersin.org/articles/10.3389/fmed.2023.1146896/full). __*Frontiers in Medicine*__. 2023, 10:1146896. (IF = 3.1，三区)


## <i class="fa fa-calendar fa-1x" aria-hidden="true"></i> 2022 {#year-2022}

+ S Xu, L Li, X Luo, M Chen, W Tang, L Zhan, Z Dai, TT. Lam, Y Guan, __G Yu__<sup>\*</sup>. [Ggtree: A serialized data object for visualization of a phylogenetic tree and annotation data](https://onlinelibrary.wiley.com/doi/full/10.1002/imt2.56). __*iMeta*__, 2022, 1(4):e56.
+ Q Wang<sup>#</sup>, M Li<sup>#</sup>, T Wu, L Zhan, L Li, M Chen, W Xie, Z Xie, E Hu, S Xu, __G Yu__<sup>\*</sup>. [Exploring epigenomic datasets by ChIPseeker](https://doi.org/10.1002/cpz1.585). __*Current Protocols*__, 2022, 2(10): e585.  [<i class="fa fa-file-pdf-o fa-2x" aria-hidden="true" style="color:black;font-size: 150%;"></i>](https://onlinelibrary.wiley.com/share/author/GYJGUBYCTRMYJFN2JFZZ?target=10.1002/cpz1.585)
+ Z Dai, T Wu, S Xu, L Zhou, W Tang, E Hu, L Zhan M Chen, __G Yu__<sup>\*</sup>. [Characterization of toxin-antitoxin systems from public sequencing data: a case study in *Pseudomonas aeruginosa*](https://www.frontiersin.org/articles/10.3389/fmicb.2022.951774). __*Frontiers in Microbiology*__. 2022, 13:951774. (IF = 5.2，二区，Top期刊)
+ T Feng<sup>#</sup>, T Wu<sup>#</sup>, Y Zhang<sup>#</sup>, L Zhou<sup>#</sup>, S Liu, L Li, M Li, E Hu, Q Wang, X Fu, L Zhan, Z Xie, W Xie, X Huang<sup>\*</sup>, X Shang<sup>\*</sup>,  __G Yu__<sup>\*</sup>. [Stemness Analysis Uncovers That The Peroxisome Proliferator-Activated Receptor Signaling Pathway Can Mediate Fatty Acid Homeostasis In Sorafenib-Resistant Hepatocellular Carcinoma Cells](https://www.frontiersin.org/articles/10.3389/fonc.2022.912694/). __*Frontiers in Oncology*__. 2022, 12:912694. (IF = 4.7，三区)
+ L Zhou<sup>#</sup>, T Feng<sup>#</sup>, S Xu, F Gao, TT Lam, Q Wang, T Wu, H Huang, L Zhan, L Li, Y Guan, Z Dai<sup>\*</sup>, __G Yu__<sup>\*</sup>. [ggmsa: a visual exploration tool for multiple sequence alignment and associated data](https://doi.org/10.1093/bib/bbac222). __*Briefings in Bioinformatics*__. 2022, 23(4):bbac222. <i class="fa fa-trophy fa-2x" aria-hidden="true" style="color:#CC5500;font-size: 150%;"></i> (IF = 9.5，一区，Top期刊)
+ N Sato, Y Tamada, __G Yu__, Y Okuno<sup>\*</sup>. [CBNplot: Bayesian network plots for enrichment analysis](https://doi.org/10.1093/bioinformatics/btac175). __*Bioinformatics*__. 2022, 38(10):2959-2960. (IF=5.8，一区，Top期刊)
+ T Wu, CL Xiao, TT Lam, __G Yu__<sup>\*</sup>. [Biomedical Data Visualization: Methods and Applications](https://www.frontiersin.org/articles/10.3389/fgene.2022.890775/full). __*Frontiers in Genetics*__. 2022, 13:890775. (IF=3.7，三区)
+ X Zhao<sup>#</sup>, W Tang<sup>#</sup>, H Wan<sup>#</sup>, Z Lan H Qin, Q Lin, Y Hu, __G Yu__<sup>\*</sup>, N Jiang<sup>\*</sup>, B Yu<sup>\*</sup>. [Altered gut microbiota as an auxiliary diagnostic indicator for patients with fracture-related infection](https://www.frontiersin.org/articles/10.3389/fmicb.2022.723791/full). __*Frontiers in Microbiology*__. 2022, 13:723791. (IF = 5.2，二区，Top期刊)


## <i class="fa fa-calendar fa-1x" aria-hidden="true"></i> 2021 {#year-2021}

+ S Xu<sup>#</sup>, M Chen<sup>#</sup>, T Feng, L Zhan, L Zhou, __G Yu__<sup>\*</sup>. [Use ggbreak to effectively utilize plotting space to deal with large datasets and outliers](https://www.frontiersin.org/articles/10.3389/fgene.2021.774846). __*Frontiers in Genetics*__. 2021, 12:774846. (IF=4.772，三区)
+ CH Gao, __G Yu__, P Cai<sup>\*</sup>. [ggVennDiagram: An intuitive, easy-to-use, and highly customizable R package to generate Venn diagram](https://doi.org/10.3389/fgene.2021.706907). __*Frontiers in Genetics*__. 2021, 12:706907. (IF=4.772，三区)
+ S Xu, Z Dai, P Guo, X Fu, S Liu, L Zhou, W Tang, T Feng, M Chen, L Zhan, T Wu, E Hu, Y Jiang<sup>\*</sup>, X Bo<sup>\*</sup>, __G Yu__<sup>\*</sup>. [ggtreeExtra: Compact visualization of richly annotated phylogenetic data](https://doi.org/10.1093/molbev/msab166). __*Molecular Biology and Evolution*__. 2021, 38(9):4039-4042. (IF=8.800，一区，Top期刊)
+ T Wu<sup>#</sup>, E Hu<sup>#</sup>, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo<sup>\*</sup>, **G Yu**<sup>\*</sup>. [clusterProfiler 4.0: A universal enrichment tool for interpreting omics data](https://doi.org/10.1016/j.xinn.2021.100141). **_The Innovation_**. 2021, 2(3):100141. 
+ D Zeng<sup>#</sup>, Z Ye<sup>#</sup>, R Shen, __G Yu__, J Wu, Y Xiong, R Zhou, W Qiu, N Huang, L Sun, X Li, J Bin, Y Liao, M Shi, W Liao<sup>\*</sup>. [IOBR: Multi-Omics Immuno-Oncology Biological Research to Decode Tumor Microenvironment and Signatures](https://doi.org/10.3389/fimmu.2021.687975). **_Frontiers in Immunology_**. 2021, 12:687975.   <i class="fa fa-trophy fa-2x" aria-hidden="true" style="color:#CC5500;font-size: 150%;"></i> (IF=8.786，二区，Top期刊)
+ T Wu, E Hu, X Ge<sup>\*</sup>, __G Yu__<sup>\*</sup>. [nCov2019: an R package for studying the COVID-19 coronavirus pandemic](https://doi.org/10.7717/peerj.11421). __*PeerJ*__. 2021, 9:e11421. (IF=3.061，三区)
+ R Huang, C Soneson, FGM Ernst, KC Rue-Albrecht, __G Yu__, SC Hicks, MD Robinson<sup>\*</sup>. [TreeSummarizedExperiment: a S4 class for data with hierarchical structure](https://doi.org/10.12688/f1000research.26669.2). __*F1000Research*__. 2021, 9:1246.
+ S Liu, __G Yu__<sup>\*</sup>, L Liu<sup>\*</sup>, X Zou, L Zhou, E Hu, Y Song. [Identification of prognostic Stromal-Immune Score-Based Genes in Hepatocellular Carcinoma microenvironment](https://www.frontiersin.org/articles/10.3389/fgene.2021.625236). __*Frontiers in Genetics*__. 2021, 12:625236. (IF=4.772，三区)


## <i class="fa fa-calendar fa-1x" aria-hidden="true"></i> 2020 {#year-2020}

+ J He<sup>#</sup>, S Xu<sup>#</sup>, B Zhang, C Xiao, Z Chen, F Si, J Fu, X Lin, G Zheng<sup>\*</sup>, __G Yu__<sup>\*</sup>, J Chen<sup>\*</sup>. [Gut microbiota and metabolite alterations associated with reduced bone mineral density or bone metabolic indexes in postmenopausal osteoporosis](https://doi.org/10.18632/aging.103168). __*Aging (Albany NY)*__. 2020, 12(9):8583-8604. (IF=5.682，二区)
+ __G Yu__<sup>\*</sup>. [Using ggtree to Visualize Data on Tree-Like Structure](https://doi.org/10.1002/cpbi.96). __*Current Protocols in Bioinformatics*__. 2020, 69(1):e96. (*__Invited paper__*) [<i class="fa fa-file-pdf-o fa-2x" aria-hidden="true" style="color:black;font-size: 150%;"></i>](https://onlinelibrary.wiley.com/share/author/WQ5Z45WFKYDM43DXQGAI?target=10.1002/cpbi.96)
+ LG Wang, TTY Lam, S Xu, Z Dai, L Zhou, T Feng, P Guo, CW Dunn, BR Jones, T Bradley, H Zhu, Y Guan, Y Jiang, __G Yu__<sup>\*</sup>. [treeio: an R package for phylogenetic tree input and output with richly annotated and associated data](https://doi.org/10.1093/molbev/msz240). __*Molecular Biology and Evolution*__. 2020, 37(2):599-603.   <i class="fa fa-trophy fa-2x" aria-hidden="true" style="color:#CC5500;font-size: 150%;"></i> (<strong>IF=16.240</strong>，一区，Top期刊)
+ Z Hao, D Lv, Y Ge, J Shi, D Weijers, __G Yu__<sup>\*</sup>, J Chen<sup>\*</sup>. [RIdeogram: drawing SVG graphics to visualize and map genome-wide data on the idiograms](https://peerj.com/articles/cs-251/). __*PeerJ Computer Science*__. 2020, 6:e251.   <i class="fa fa-trophy fa-2x" aria-hidden="true" style="color:#CC5500;font-size: 150%;"></i> (IF=1.392，三区)


## <i class="fa fa-calendar fa-1x" aria-hidden="true"></i> 2018 {#year-2018}


+ __G Yu__<sup>\*</sup>, TTY Lam, H Zhu, Y Guan<sup>\*</sup>. [Two methods for mapping and visualizing associated data on phylogeny using ggtree](https://academic.oup.com/mbe/article-abstract/35/12/3041/5142656). __*Molecular Biology and Evolution*__. 2018, 35(12):3041-3043.   <i class="fa fa-trophy fa-2x" aria-hidden="true" style="color:#CC5500;font-size: 150%;"></i> (<strong>IF=14.797</strong>，一区， Top期刊)
+ __G Yu__<sup>\*</sup>. [Using meshes for MeSH term enrichment and semantic analyses](https://doi.org/10.1093/bioinformatics/bty410). __*Bioinformatics*__. 2018, 34(21):3766-3767. (IF=4.531, 一区， Top期刊)


## <i class="fa fa-bar-chart fa-1x" aria-hidden="true"></i> Selected publications (2010-2017) {#year-2010-2017}

+ __G Yu__, DK Smith, H Zhu, Y Guan, TTY Lam<sup>\*</sup>. [ggtree: an R package for visualization and annotation of phylogenetic trees with their covariates and other associated data](http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/abstract). __*Methods in Ecology and Evolution*__. 2017, 8(1):28-36.  <i class="fa fa-trophy fa-2x" aria-hidden="true" style="color:#CC5500;font-size: 150%;"></i> (IF=6.363, 一区，Top期刊)
+ __G Yu__, QY He<sup>\*</sup>. [ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization](http://pubs.rsc.org/en/Content/ArticleLanding/2016/MB/C5MB00663E). __*Molecular BioSystems*__. 2016, 12(2):477-479.  <i class="fa fa-trophy fa-2x" aria-hidden="true" style="color:#CC5500;font-size: 150%;"></i> (IF=2.781，三区)
+ __G Yu__<sup>\*</sup>, LG Wang, QY He<sup>\*</sup>. [ChIPseeker: an R/Bioconductor package for ChIP peak annotation, comparision and visualization](http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btv145). __*Bioinformatics*__. 2015, 31(14):2382-2383.   <i class="fa fa-trophy fa-2x" aria-hidden="true" style="color:#CC5500;font-size: 150%;"></i> (IF=5.766, 一区， Top期刊)
+	__G Yu__<sup>\*</sup>, LG Wang, GR Yan, QY He<sup>\*</sup>. [DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis](http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btu684). __*Bioinformatics*__. 2015, 31(4):608-609.   <i class="fa fa-trophy fa-2x" aria-hidden="true" style="color:#CC5500;font-size: 150%;"></i> (IF=5.766, 一区， Top期刊)
+	__G Yu__, LG Wang, Y Han, QY He<sup>\*</sup>. [clusterProfiler: an R package for comparing biological themes among gene clusters](http://online.liebertpub.com/doi/abs/10.1089/omi.2011.0118). __*OMICS: A Journal of Integrative Biology*__. 2012, 16(5):284-287. (IF=2.730， 四区)
+ __G Yu__<sup>#</sup>, F Li<sup>#</sup>, Y Qin, X Bo<sup>\*</sup>, Y Wu, S Wang<sup>\*</sup>. [GOSemSim: an R package for measuring semantic similarity among GO terms and gene products](http://bioinformatics.oxfordjournals.org/cgi/content/abstract/26/7/976). __*Bioinformatics*__. 2010, 26(7):976-978. (IF=4.877, 二区， Top期刊)


## <i class="fa fa-book fa-1x" aria-hidden="true"></i> Books {#books}

+ 沈百荣，杨恩策. 基于理工信的医学数据采集与分析（基础医学“101计划”核心教材）. __*北京大学医学出版社*__, 2024. ISBN: [9787565931987](https://www.pumpedu.com/en-shop/7263.html)
+ __余光创（著）__, 李林, 罗晓（译）. R实战：系统发育树的数据集成操作及可视化. __*电子工业出版社*__, 2023. ISBN: [9787121451829](https://u.jd.com/uu7G4qC) 
+ __G Yu__. Data Integration, Manipulation and Visualization of Phylogenetic Treess (1st edition). __*Chapman and Hall/CRC*__, 2023. doi: [10.1201/9781003279242](https://doi.org/10.1201/9781003279242), ISBN: [9781032233574](https://www.amazon.com/Integration-Manipulation-Visualization-Phylogenetic-Computational/dp/1032233575)
+ **Yu G**. [Gene Ontology Semantic Similarity Analysis Using GOSemSim](http://dx.doi.org/10.1007/978-1-0716-0301-7_11). In: Kidder B. (eds) Stem Cell Transcriptional Networks. ***Methods in Molecular Biology***, 2020, 2117:207-215. Humana, New York, NY. ISBN: [9781071603000](https://link.springer.com/book/10.1007/978-1-0716-0301-7)

