<!DOCTYPE html>
<html lang="en">

<head>
  <title>
  Software Â· YuLab@SMU
</title>
  <meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="color-scheme" content="light dark">




<meta name="author" content="YuLab-SMU">
<meta name="description" content="Infectious disease and Epidemiology Link to heading nCov2019: An R package for exploring COVID-19 statistics seqcombo: Visualization Tool for Sequence Recombination and Reassortment Knowledge mining Link to heading clusterProfiler: Statistical analysis and visualization of functional profiles for genes and gene clusters DOSE: Disease Ontology Semantic and Enrichment analysis enrichplot: Visualization of Functional Enrichment Result gson: Base Class and Methods for &lsquo;gson&rsquo; Format GOSemSim: GO-terms Semantic Similarity Measures HDO.db: A set of annotation maps describing the entire Human Disease Ontology meshes: MeSH Enrichment and Semantic analyses ReactomePA: Reactome Pathway Analysis Microbiome Link to heading MicrobiotaProcess: Analysis, visualization and biomarker discovery of microbiome MMINP: Microbe-Metabolite INteractions-based metabolic profiles Predictor OMICS data analysis Link to heading ChIPseeker: ChIP peak Annotation, Comparison, and Visualization GeoTcgaData: Processing various types of data on GEO and TCGA Sequence analysis Link to heading ggmsa: Plot mutiple sequence alignment using &lsquo;ggplot2&rsquo; seqmagick: Sequence Manipulation Utilities Tree Data integration and visualization Link to heading ggtree: Visualization of tree and annotation data ggtreeDendro: Drawing &lsquo;dendrogram&rsquo; using &lsquo;ggtree&rsquo; ggtreeExtra: Phylogenetic tree data annotation shinyTempSignal: Explore Temporal Signal of Molecular Phylogenies tidytree: A Tidy Tool for Phylogenetic Tree Data Manipulation treeio: Base Classes and Functions for Phylogenetic Tree Input and Output Visualization Link to heading aplot: Decorating a plot with associated information emojifont: Emoji and Font Awesome in Graphics ggbreak: An implementation of scale functions for setting axis breaks of a &lsquo;gg&rsquo; plot.">
<meta name="keywords" content="developer,bioinformatics,omics,visualization">

<meta name="twitter:card" content="summary"/><meta name="twitter:title" content="Software"/>
<meta name="twitter:description" content="Infectious disease and Epidemiology Link to heading nCov2019: An R package for exploring COVID-19 statistics seqcombo: Visualization Tool for Sequence Recombination and Reassortment Knowledge mining Link to heading clusterProfiler: Statistical analysis and visualization of functional profiles for genes and gene clusters DOSE: Disease Ontology Semantic and Enrichment analysis enrichplot: Visualization of Functional Enrichment Result gson: Base Class and Methods for &lsquo;gson&rsquo; Format GOSemSim: GO-terms Semantic Similarity Measures HDO.db: A set of annotation maps describing the entire Human Disease Ontology meshes: MeSH Enrichment and Semantic analyses ReactomePA: Reactome Pathway Analysis Microbiome Link to heading MicrobiotaProcess: Analysis, visualization and biomarker discovery of microbiome MMINP: Microbe-Metabolite INteractions-based metabolic profiles Predictor OMICS data analysis Link to heading ChIPseeker: ChIP peak Annotation, Comparison, and Visualization GeoTcgaData: Processing various types of data on GEO and TCGA Sequence analysis Link to heading ggmsa: Plot mutiple sequence alignment using &lsquo;ggplot2&rsquo; seqmagick: Sequence Manipulation Utilities Tree Data integration and visualization Link to heading ggtree: Visualization of tree and annotation data ggtreeDendro: Drawing &lsquo;dendrogram&rsquo; using &lsquo;ggtree&rsquo; ggtreeExtra: Phylogenetic tree data annotation shinyTempSignal: Explore Temporal Signal of Molecular Phylogenies tidytree: A Tidy Tool for Phylogenetic Tree Data Manipulation treeio: Base Classes and Functions for Phylogenetic Tree Input and Output Visualization Link to heading aplot: Decorating a plot with associated information emojifont: Emoji and Font Awesome in Graphics ggbreak: An implementation of scale functions for setting axis breaks of a &lsquo;gg&rsquo; plot."/>

<meta property="og:title" content="Software" />
<meta property="og:description" content="Infectious disease and Epidemiology Link to heading nCov2019: An R package for exploring COVID-19 statistics seqcombo: Visualization Tool for Sequence Recombination and Reassortment Knowledge mining Link to heading clusterProfiler: Statistical analysis and visualization of functional profiles for genes and gene clusters DOSE: Disease Ontology Semantic and Enrichment analysis enrichplot: Visualization of Functional Enrichment Result gson: Base Class and Methods for &lsquo;gson&rsquo; Format GOSemSim: GO-terms Semantic Similarity Measures HDO.db: A set of annotation maps describing the entire Human Disease Ontology meshes: MeSH Enrichment and Semantic analyses ReactomePA: Reactome Pathway Analysis Microbiome Link to heading MicrobiotaProcess: Analysis, visualization and biomarker discovery of microbiome MMINP: Microbe-Metabolite INteractions-based metabolic profiles Predictor OMICS data analysis Link to heading ChIPseeker: ChIP peak Annotation, Comparison, and Visualization GeoTcgaData: Processing various types of data on GEO and TCGA Sequence analysis Link to heading ggmsa: Plot mutiple sequence alignment using &lsquo;ggplot2&rsquo; seqmagick: Sequence Manipulation Utilities Tree Data integration and visualization Link to heading ggtree: Visualization of tree and annotation data ggtreeDendro: Drawing &lsquo;dendrogram&rsquo; using &lsquo;ggtree&rsquo; ggtreeExtra: Phylogenetic tree data annotation shinyTempSignal: Explore Temporal Signal of Molecular Phylogenies tidytree: A Tidy Tool for Phylogenetic Tree Data Manipulation treeio: Base Classes and Functions for Phylogenetic Tree Input and Output Visualization Link to heading aplot: Decorating a plot with associated information emojifont: Emoji and Font Awesome in Graphics ggbreak: An implementation of scale functions for setting axis breaks of a &lsquo;gg&rsquo; plot." />
<meta property="og:type" content="article" />
<meta property="og:url" content="http://yulab-smu.top/software/" /><meta property="article:section" content="" />








<link rel="canonical" href="http://yulab-smu.top/software/">


<link rel="preload" href="/fonts/forkawesome-webfont.woff2?v=1.2.0" as="font" type="font/woff2" crossorigin>


  
  
  <link rel="stylesheet" href="/css/coder.min.135e22c97ff685fe983fc60048e309ced8f00d8d38f536aa67dba8a13a03dfa4.css" integrity="sha256-E14iyX/2hf6YP8YASOMJztjwDY049TaqZ9uooToD36Q=" crossorigin="anonymous" media="screen" />








 




<link rel="icon" type="image/svg+xml" href="/images/favicon.svg" sizes="any">
<link rel="icon" type="image/png" href="/images/favicon-32x32.png" sizes="32x32">
<link rel="icon" type="image/png" href="/images/favicon-16x16.png" sizes="16x16">

<link rel="apple-touch-icon" href="/images/apple-touch-icon.png">
<link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon.png">

<link rel="manifest" href="/site.webmanifest">
<link rel="mask-icon" href="/images/safari-pinned-tab.svg" color="#5bbad5">









</head>




<body class="preload-transitions colorscheme-light">
  
<div class="float-container">
    <a id="dark-mode-toggle" class="colorscheme-toggle">
        <i class="fa fa-adjust fa-fw" aria-hidden="true"></i>
    </a>
</div>


  <main class="wrapper">
    <nav class="navigation">
  <section class="container">
    <a class="navigation-title" href="/">
      YuLab@SMU
    </a>
    
      <input type="checkbox" id="menu-toggle" />
      <label class="menu-button float-right" for="menu-toggle">
        <i class="fa fa-bars fa-fw" aria-hidden="true"></i>
      </label>
      <ul class="navigation-list">
        
          
            <li class="navigation-item">
              <a class="navigation-link" href="/about/">About</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/publications/">Publications</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/contributions/">Contributions</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/members/">Members</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/news/">News</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/contact/">Contact</a>
            </li>
          
        
        
      </ul>
    
  </section>
</nav>


    <div class="content">
      
  <section class="container page">
  <article>
    <header>
      <h1 class="title">
        <a class="title-link" href="http://yulab-smu.top/software/">
          Software
        </a>
      </h1>
    </header>

    <h2 id="i-classfa-fa-plus-square-fa-1x-aria-hiddentruei-infectious-disease-and-epidemiology">
  <i class="fa fa-plus-square fa-1x" aria-hidden="true"></i> Infectious disease and Epidemiology
  <a class="heading-link" href="#i-classfa-fa-plus-square-fa-1x-aria-hiddentruei-infectious-disease-and-epidemiology">
    <i class="fa fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<ul>
<li><a href="https://github.com/YuLab-SMU/nCov2019"  class="external-link" target="_blank" rel="noopener">nCov2019</a>: An R package for exploring COVID-19 statistics</li>
<li><a href="https://www.bioconductor.org/packages/seqcombo"  class="external-link" target="_blank" rel="noopener">seqcombo</a>: Visualization Tool for Sequence Recombination and Reassortment</li>
</ul>
<h2 id="i-classfa-fa-yelp-fa-1x-aria-hiddentruei-knowledge-mining">
  <i class="fa fa-yelp fa-1x" aria-hidden="true"></i> Knowledge mining
  <a class="heading-link" href="#i-classfa-fa-yelp-fa-1x-aria-hiddentruei-knowledge-mining">
    <i class="fa fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<ul>
<li><a href="https://www.bioconductor.org/packages/clusterProfiler"  class="external-link" target="_blank" rel="noopener">clusterProfiler</a>: Statistical analysis and visualization of functional profiles for genes and gene clusters</li>
<li><a href="https://www.bioconductor.org/packages/DOSE"  class="external-link" target="_blank" rel="noopener">DOSE</a>: Disease Ontology Semantic and Enrichment analysis</li>
<li><a href="https://www.bioconductor.org/packages/enrichplot"  class="external-link" target="_blank" rel="noopener">enrichplot</a>: Visualization of Functional Enrichment Result</li>
<li><a href="https://cran.r-project.org/package=gson"  class="external-link" target="_blank" rel="noopener">gson</a>: Base Class and Methods for &lsquo;gson&rsquo; Format</li>
<li><a href="https://www.bioconductor.org/packages/GOSemSim"  class="external-link" target="_blank" rel="noopener">GOSemSim</a>: GO-terms Semantic Similarity Measures</li>
<li><a href="https://www.bioconductor.org/packages/HDO.db"  class="external-link" target="_blank" rel="noopener">HDO.db</a>: A set of annotation maps describing the entire Human Disease Ontology</li>
<li><a href="https://www.bioconductor.org/packages/meshes"  class="external-link" target="_blank" rel="noopener">meshes</a>: MeSH Enrichment and Semantic analyses</li>
<li><a href="https://www.bioconductor.org/packages/ReactomePA"  class="external-link" target="_blank" rel="noopener">ReactomePA</a>: Reactome Pathway Analysis</li>
</ul>
<h2 id="i-classfa-fa-shaarli-o-fa-1x-aria-hiddentruei-microbiome">
  <i class="fa fa-shaarli-o fa-1x" aria-hidden="true"></i> Microbiome
  <a class="heading-link" href="#i-classfa-fa-shaarli-o-fa-1x-aria-hiddentruei-microbiome">
    <i class="fa fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<ul>
<li><a href="https://www.bioconductor.org/packages/MicrobiotaProcess"  class="external-link" target="_blank" rel="noopener">MicrobiotaProcess</a>: Analysis, visualization and biomarker discovery of microbiome</li>
<li><a href="https://cran.r-project.org/package=MMINP"  class="external-link" target="_blank" rel="noopener">MMINP</a>: Microbe-Metabolite INteractions-based metabolic profiles Predictor</li>
</ul>
<h2 id="i-classfa-fa-bullseye-fa-1x-aria-hiddentruei-omics-data-analysis">
  <i class="fa fa-bullseye fa-1x" aria-hidden="true"></i> OMICS data analysis
  <a class="heading-link" href="#i-classfa-fa-bullseye-fa-1x-aria-hiddentruei-omics-data-analysis">
    <i class="fa fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<ul>
<li><a href="https://www.bioconductor.org/packages/ChIPseeker"  class="external-link" target="_blank" rel="noopener">ChIPseeker</a>: ChIP peak Annotation, Comparison, and Visualization</li>
<li><a href="https://cran.r-project.org/package=GeoTcgaData"  class="external-link" target="_blank" rel="noopener">GeoTcgaData</a>: Processing various types of data on GEO and TCGA</li>
</ul>
<h2 id="i-classfa-fa-line-chart-fa-1x-aria-hiddentruei-sequence-analysis">
  <i class="fa fa-line-chart fa-1x" aria-hidden="true"></i> Sequence analysis
  <a class="heading-link" href="#i-classfa-fa-line-chart-fa-1x-aria-hiddentruei-sequence-analysis">
    <i class="fa fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<ul>
<li><a href="https://cran.r-project.org/package=ggmsa"  class="external-link" target="_blank" rel="noopener">ggmsa</a>: Plot mutiple sequence alignment using &lsquo;ggplot2&rsquo;</li>
<li><a href="https://cran.r-project.org/package=seqmagick"  class="external-link" target="_blank" rel="noopener">seqmagick</a>: Sequence Manipulation Utilities</li>
</ul>
<p><a href="https://www.routledge.com/Data-Integration-Manipulation-and-Visualization-of-Phylogenetic-Trees/Yu/p/book/9781032233574"><img src="https://yulab-smu.top/treedata-book/9781032233574_cover_review.png" style="width:200px;border:2px solid black;" align="right"/></a></p>
<h2 id="i-classfa-fa-tree-fa-1x-aria-hiddentruei-tree-data-integration-and-visualization">
  <i class="fa fa-tree fa-1x" aria-hidden="true"></i> Tree Data integration and visualization
  <a class="heading-link" href="#i-classfa-fa-tree-fa-1x-aria-hiddentruei-tree-data-integration-and-visualization">
    <i class="fa fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<ul>
<li><a href="https://www.bioconductor.org/packages/ggtree"  class="external-link" target="_blank" rel="noopener">ggtree</a>: Visualization of tree and annotation data</li>
<li><a href="https://www.bioconductor.org/packages/ggtreeDendro"  class="external-link" target="_blank" rel="noopener">ggtreeDendro</a>: Drawing &lsquo;dendrogram&rsquo; using &lsquo;ggtree&rsquo;</li>
<li><a href="https://www.bioconductor.org/packages/ggtreeExtra"  class="external-link" target="_blank" rel="noopener">ggtreeExtra</a>: Phylogenetic tree data annotation</li>
<li><a href="https://cran.r-project.org/package=shinyTempSignal"  class="external-link" target="_blank" rel="noopener">shinyTempSignal</a>: Explore Temporal Signal of Molecular Phylogenies</li>
<li><a href="https://cran.r-project.org/package=tidytree"  class="external-link" target="_blank" rel="noopener">tidytree</a>: A Tidy Tool for Phylogenetic Tree Data Manipulation</li>
<li><a href="https://www.bioconductor.org/packages/"  class="external-link" target="_blank" rel="noopener">treeio</a>: Base Classes and Functions for Phylogenetic Tree Input and Output</li>
</ul>
<h2 id="i-classfa-fa-pie-chart-fa-1x-aria-hiddentruei-visualization">
  <i class="fa fa-pie-chart fa-1x" aria-hidden="true"></i> Visualization
  <a class="heading-link" href="#i-classfa-fa-pie-chart-fa-1x-aria-hiddentruei-visualization">
    <i class="fa fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<ul>
<li><a href="https://cran.r-project.org/package=aplot"  class="external-link" target="_blank" rel="noopener">aplot</a>: Decorating a plot with associated information</li>
<li><a href="https://cran.r-project.org/package=emojifont"  class="external-link" target="_blank" rel="noopener">emojifont</a>: Emoji and Font Awesome in Graphics</li>
<li><a href="https://cran.r-project.org/package=ggbreak"  class="external-link" target="_blank" rel="noopener">ggbreak</a>: An implementation of scale functions for setting axis breaks of a &lsquo;gg&rsquo; plot.</li>
<li><a href="https://cran.r-project.org/package=ggimage"  class="external-link" target="_blank" rel="noopener">ggimage</a>: Use Image in &lsquo;ggplot2&rsquo;</li>
<li><a href="https://cran.r-project.org/package=ggplotify"  class="external-link" target="_blank" rel="noopener">ggplotify</a>: Convert Plot to &lsquo;grob&rsquo; or &lsquo;ggplot&rsquo; Object</li>
<li><a href="https://cran.r-project.org/package=ggstar"  class="external-link" target="_blank" rel="noopener">ggstar</a>: Multiple Geometric Shape Point Layer for &lsquo;ggplot2&rsquo;</li>
<li><a href="https://cran.r-project.org/package=meme"  class="external-link" target="_blank" rel="noopener">meme</a>: Create Meme</li>
<li><a href="https://cran.r-project.org/package=plotbb"  class="external-link" target="_blank" rel="noopener">plotbb</a>: Grammar of Graphics for base plot</li>
<li><a href="https://cran.r-project.org/package=scatterpie"  class="external-link" target="_blank" rel="noopener">scatterpie</a>: Creates scatterpie plots, especially useful for plotting pies on a map</li>
</ul>

  </article>
</section>

  

    </div>

    <footer class="footer">
  <section class="container">
    Â©
    
      2018 -
    
    2024
     YuLab-SMU 
    Â·
    
    Powered by <a href="https://gohugo.io/" target="_blank" rel="noopener">Hugo</a> & <a href="https://github.com/luizdepra/hugo-coder/" target="_blank" rel="noopener">Coder</a>.
    
  </section>
</footer>

  </main>

  

  
  
  <script src="/js/coder.min.6ae284be93d2d19dad1f02b0039508d9aab3180a12a06dcc71b0b0ef7825a317.js" integrity="sha256-auKEvpPS0Z2tHwKwA5UI2aqzGAoSoG3McbCw73gloxc="></script>
  

  

  


  

  

  

  <script data-goatcounter="https://code.goatcounter.com/count"
        async src="//gc.zgo.at/count.js"></script>


  
<script defer src='https://static.cloudflareinsights.com/beacon.min.js'
        data-cf-beacon='{"token": "token"}'></script>



  

  

  

  

  

  

  

  

  
</body>

</html>
