<!DOCTYPE html>
<html lang="en">

<head>
  <title>
  Tree Data Integration · YuLab@SMU
</title>
  <meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="color-scheme" content="light dark">




<meta name="author" content="YuLab-SMU">
<meta name="description" content="A series of methods and software tools have been proposed and developed for the operation, integration, and visualization of trees and data. This includes (1) the introduction of graphic grammar into the field of phylogeny for the first time, (2) enhancing the data integration ability of phylogenetics and its application in different disciplines, (3) proposing two universal methods for phylogenetic data integration and visualization, and (4) designing data structures that can store phylogenetic trees, related data, and visualization directives to ensure analytical reproducibility.">
<meta name="keywords" content="developer,bioinformatics,omics,visualization">

<meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="Tree Data Integration"/>
<meta name="twitter:description" content="A series of methods and software tools have been proposed and developed for the operation, integration, and visualization of trees and data. This includes (1) the introduction of graphic grammar into the field of phylogeny for the first time, (2) enhancing the data integration ability of phylogenetics and its application in different disciplines, (3) proposing two universal methods for phylogenetic data integration and visualization, and (4) designing data structures that can store phylogenetic trees, related data, and visualization directives to ensure analytical reproducibility."/>

<meta property="og:title" content="Tree Data Integration" />
<meta property="og:description" content="A series of methods and software tools have been proposed and developed for the operation, integration, and visualization of trees and data. This includes (1) the introduction of graphic grammar into the field of phylogeny for the first time, (2) enhancing the data integration ability of phylogenetics and its application in different disciplines, (3) proposing two universal methods for phylogenetic data integration and visualization, and (4) designing data structures that can store phylogenetic trees, related data, and visualization directives to ensure analytical reproducibility." />
<meta property="og:type" content="article" />
<meta property="og:url" content="http://yulab-smu.top/achievement-tree-data/" /><meta property="article:section" content="" />







<link rel="canonical" href="http://yulab-smu.top/achievement-tree-data/">


<link rel="preload" href="/fonts/forkawesome-webfont.woff2?v=1.2.0" as="font" type="font/woff2" crossorigin>


  
  
  <link rel="stylesheet" href="/css/coder.min.135e22c97ff685fe983fc60048e309ced8f00d8d38f536aa67dba8a13a03dfa4.css" integrity="sha256-E14iyX/2hf6YP8YASOMJztjwDY049TaqZ9uooToD36Q=" crossorigin="anonymous" media="screen" />








 




<link rel="icon" type="image/svg+xml" href="/images/favicon.svg" sizes="any">
<link rel="icon" type="image/png" href="/images/favicon-32x32.png" sizes="32x32">
<link rel="icon" type="image/png" href="/images/favicon-16x16.png" sizes="16x16">

<link rel="apple-touch-icon" href="/images/apple-touch-icon.png">
<link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon.png">

<link rel="manifest" href="/site.webmanifest">
<link rel="mask-icon" href="/images/safari-pinned-tab.svg" color="#5bbad5">









</head>




<body class="preload-transitions colorscheme-light">
  
<div class="float-container">
    <a id="dark-mode-toggle" class="colorscheme-toggle">
        <i class="fa fa-adjust fa-fw" aria-hidden="true"></i>
    </a>
</div>


  <main class="wrapper">
    <nav class="navigation">
  <section class="container">
    <a class="navigation-title" href="/">
      YuLab@SMU
    </a>
    
      <input type="checkbox" id="menu-toggle" />
      <label class="menu-button float-right" for="menu-toggle">
        <i class="fa fa-bars fa-fw" aria-hidden="true"></i>
      </label>
      <ul class="navigation-list">
        
          
            <li class="navigation-item">
              <a class="navigation-link" href="/about/">About</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/publications/">Publications</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/software/">Software</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/members/">Members</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/news/">News</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/contact/">Contact</a>
            </li>
          
        
        
      </ul>
    
  </section>
</nav>


    <div class="content">
      
  <section class="container page">
  <article>
    <header>
      <h1 class="title">
        <a class="title-link" href="http://yulab-smu.top/achievement-tree-data/">
          Tree Data Integration
        </a>
      </h1>
    </header>

    <style>
    @media only screen and (max-width: 768px) {
        td {
            display: block;
        }
    }
</style>
<p>A series of methods and software tools have been proposed and developed for the operation, integration, and visualization of trees and data. This includes (1) the introduction of graphic grammar into the field of phylogeny for the first time, (2) enhancing the data integration ability of phylogenetics and its application in different disciplines, (3) proposing two universal methods for phylogenetic data integration and visualization, and (4) designing data structures that can store phylogenetic trees, related data, and visualization directives to ensure analytical reproducibility. This series of methods and software tools provide a concise and unified syntax as a whole system., which assists researchers in discovering hidden patterns and proposing new hypotheses by integrating heterogeneous data in the context of evolution or hierachy.</p>
<p>Two monographs have been published to introduce this series of work, including &ldquo;<a href="https://www.routledge.com/Data-Integration-Manipulation-and-Visualization-of-Phylogenetic-Trees/Yu/p/book/9781032233574"  class="external-link" target="_blank" rel="noopener">Data integration, manipulation and visualization of phylogenetic trees</a>&rdquo; (EN) published by CRC Press and 《<a href="https://weread.qq.com/web/bookDetail/8ad32a00813ab81bbg0183d2"  class="external-link" target="_blank" rel="noopener">R实战：系统发育树的数据集成操作与可视化</a>》 (CN) published by Publishing house of electronics industry (电子工业出版社).</p>
<p><a href="https://www.routledge.com/Data-Integration-Manipulation-and-Visualization-of-Phylogenetic-Trees/Yu/p/book/9781032233574"><img src="https://yulab-smu.top/treedata-book/9781032233574_cover_review.png" style="height:320px;border:2px solid black;"/></a>
<a href="https://weread.qq.com/web/bookDetail/8ad32a00813ab81bbg0183d2"><img src="/images/book-cover-cn.png" style="height:320px;border:2px solid black;"/></a></p>
<hr>
<h2 id="1-frist-time-implementation-of-graphical-grammar-in-visualizing-phylogenetic-tree-data">
  1. Frist-time implementation of graphical grammar in visualizing phylogenetic tree data
  <a class="heading-link" href="#1-frist-time-implementation-of-graphical-grammar-in-visualizing-phylogenetic-tree-data">
    <i class="fa fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:25%">
<p><a href="http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/abstract"><img src="/images/ggtree/ggtree-2017.png" width='1000px'/></a></p>
</td>
<td style="border:none;">
<p>There are numerous software implementations for phylogenetic tree visualization, yet they are mostly used to visualize the tree&rsquo;s topological structure and are unable to (or only have limited support for) annotating the tree using knowledge or data. For the first time, <a href="https://www.bioconductor.org/packages/ggtree"  class="external-link" target="_blank" rel="noopener">ggtree</a> introduces the grammar of graphics into the visualization of phylogenetic trees and related data, effectively implementing a high level of abstraction for visualization through simple grammar, greatly reducing the difficulty of data visualization and making complex requirements possible. This work was published in <a href="http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/abstract"  class="external-link" target="_blank" rel="noopener"><em>Methods in Ecology and Evolution 2017</em></a> (ESI highly cited), and was selected by the journal as <a href="https://methodsblog.com/2020/11/19/ggtree-tree-visualization/"  class="external-link" target="_blank" rel="noopener">one of the ten representative works</a> for its 10th anniversary celebration. A protocol paper (invited) demonstrating the use of this package was published in <a href="https://currentprotocols.onlinelibrary.wiley.com/doi/abs/10.1002/cpbi.96"  class="external-link" target="_blank" rel="noopener"><em>Current Protocols in Bioinformatics 2020</em></a>.</p>
</td>
</tr>
</table>
<!--
系统发育树可视化有非常多的软件实现，然而主要用于可视化树的拓扑结构，无法（或仅有限支持）利用知识和数据对树进行注释，ggtree首次将图形语法引入到系统发育树及相关数据的可视化，有效地实现了简单语法对可视化需求的高度抽象，大大降低了数据可视化的难度，让复杂的需求成为可能.该工作发表于 Methods in Ecology and Evolution 2017, 该论文被期刊选为 10 周年纪念 10 篇代表作之一。
-->
<h2 id="2-base-classes-and-functions-for-phylogenetic-tree-input-and-output">
  2. Base classes and functions for phylogenetic tree input and output
  <a class="heading-link" href="#2-base-classes-and-functions-for-phylogenetic-tree-input-and-output">
    <i class="fa fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:25%">
<p><a href="https://academic.oup.com/mbe/article/37/2/599/5601621"><img src="/images/ggtree/treeio-2020.png" width='1000px'/></a></p>
</td>
<td style="border:none;">
<p>The outputs of phylogenetic software are mostly in non-standard formats and not compatible with each other. This restricts the integration and comparative analysis in downstream applications. To address this issue, we developed <a href="https://www.bioconductor.org/packages/treeio"  class="external-link" target="_blank" rel="noopener">treeio</a>, which allows for parsing of standard and a dozen of non-standard data formats, and enables integration of external data. It also supports exporting the phylogenetic tree and related data into a single file. The ability to support input and output means that data format conversion is possible, thereby making more software indirectly support a wider range of data. The parsing and integration of various types of data provides the potential for downstream integrated analyses and comparative analyses, while expanding the scope of application of phylogenetic analysis. This work has been published in <a href="https://academic.oup.com/mbe/article/37/2/599/5601621"  class="external-link" target="_blank" rel="noopener"><em>Molecular Biology and Evolution 2020</em></a> (ESI highly cited).</p>
</td>
</tr>
</table>
<!--
系统发育相关的软件输出多数为非标准格式，互相之间无法解析，使得分析结果无法在不同的软件中使用，极大地限制了下游的整合与比较分析。针对这一问题，我们开发了基础的输入输出软件工具 － treeio，允许解析标准的数据格式以及十几种非标准的软件输出，并且允许整合外部数据。此外还支持将系统发育树与相关数据输出到单一的文件中。输入和输出的支持，意味着允许数据进行格式转换，进而让更多的软件变相地支持更多的数据。而多种数据的解析和整合，则为下游的整合分析、比较分析提供了可能性，也扩展了系统发育分析的应用范畴。将该工作发表于 Molecular Biology and Evolution 2020。
-->
<h2 id="3-propose-two-general-methods-for-the-integration-and-visualization-of-phylogenetic-data">
  3. Propose two general methods for the integration and visualization of phylogenetic data
  <a class="heading-link" href="#3-propose-two-general-methods-for-the-integration-and-visualization-of-phylogenetic-data">
    <i class="fa fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:25%">
<p><a href="https://doi.org/10.1093/molbev/msab166"><img src="/images/ggtree/ggtreeExtra-2021.png" width='1000px'/></a></p>
</td>
<td style="border:none;">
<p>Two general methods have been proposed and implemented, covering all aspects of the integration and visualization of phylogenetic data. The first method allows data to be mapped to the topology of the tree and supports the direct display of data or its mapping as visualization features; the second method reorganizes external data according to the tree topology, visualizes it in the manner specified by the user, and finally aligns the visualization result with the phylogenetic tree. These two general methods allow various heterogeneous data from different disciplines to be deciphered in the context of phylogenetics, which can contribute to the discovery of new patterns related or the proposal of new hypotheses. This work was published in <a href="https://academic.oup.com/mbe/article/35/12/3041/5142656"  class="external-link" target="_blank" rel="noopener"><em>Molecular Biology and Evolution in 2018</em></a> (ESI highly cited). Based on this, the <a href="https://www.bioconductor.org/packages/ggtreeExtra"  class="external-link" target="_blank" rel="noopener">ggtreeExtra</a> package was developed to enhance the integration and visualization capabilities for richly annotated data and was published in <a href="https://doi.org/10.1093/molbev/msab166"  class="external-link" target="_blank" rel="noopener"><em>Molecular Biology and Evolution in 2021</em></a>.</p>
</td>
</tr>
</table>
<!-- 提出并实现了两种通用的方法，涵盖了系统发育数据整合与可视化的方方面面。方法一允许将数据映射到树的拓扑结构上，并支持将数据直接展示或映射为可视化特征（如颜色、大小、粗细等）；方法二则将外部数据按照树的拓扑结构进行重排，再按照用户指定的方式进行可视化，最终将可视化结果与系统发育树对齐展示，方便研究人员结合系统发育信息对数据进行解读。两种通用的方法使得来自于不同学科的各种异质性的数据能够在系统发育的背景下得以解析，有助于发现与演化相关的新模式或提出新的假说。将该工作发表于Molecular Biology and Evolution 2018。在此基础上，又实现了ggtreeExtra包，以加强高维度数据的整合可视化能力，这一工作发表于Molecular Biology and Evolution 2021。
--> 
<h2 id="4-enhance-data-reuse-and-analytical-reproducibility">
  4. Enhance data reuse and analytical reproducibility
  <a class="heading-link" href="#4-enhance-data-reuse-and-analytical-reproducibility">
    <i class="fa fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:25%">
<p><a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/imt2.56"><img src="/images/ggtree/imeta-2022.png" width='1000px'/></a></p>
</td>
<td style="border:none;">
<p>Visualisation of phylogenetic trees is usually in the form of images, and the corresponding trees and data cannot be reused, which hampers the integration of phylogenetic knowledge and comparative analysis. Studies have shown that about 60% of the published phylogenetic data is permanently lost. To solve this problem, we designed the <em>ggtree</em> object, which encompasses the phylogenetic tree, data, and visualization directives. It can be rendered into an image, while the phylogenetic tree and related data can be extracted from it. In addition, similar to the &ldquo;format painter&rdquo;, the visualization directives can be used for visualizing other tree objects. This work was published in <a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/imt2.56"  class="external-link" target="_blank" rel="noopener"><em>iMeta 2022</em></a>, supporting data reusability and research replicability while promoting the integration and comparative analysis of phylogenetic data in the field.</p>
</td>
</tr>
</table>
<!--

系统发育树的可视化通常以图片的形式，而相应的树和数据无法被重新使用，以整合系统发育学知识以及进行比较分析，有研究表明，约 60%的已发表系统发育数据永久地丢失了。为了解决这一问题，我们设计了 ggtree 对象，包含了系统发育树、数据和可视化指令，它可以渲染成一张图片，同时能够从中抽提出系统发育树和相关的数据，并且类似于“格式刷”，可视化指令还能用于可视化其它的树对象。这一工作发表于iMeta 2022，有力地支持数据的可重用性和研究的可重复性，同时也能够促进学科对系统发育数据的整合与比较分析。
-->
<h2 id="5-extend-support-for-other-tree-like-structures">
  5. Extend support for other tree-like structures
  <a class="heading-link" href="#5-extend-support-for-other-tree-like-structures">
    <i class="fa fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:25%">
<p><a href="https://www.bioconductor.org/packages/ggtreeDendro"><img src="/images/ggtree/ggtreeDendro.png" width='1000px'/></a></p>
</td>
<td style="border:none;">
<p>Expand the series of tools related to tree data integration and visualization to apply to other tree-like structures (e.g., hierarchical clustering and classification/regression trees). Implemented the <a href="https://www.bioconductor.org/packages/ggtreeDendro"  class="external-link" target="_blank" rel="noopener">ggtreeDendro</a> package to support general hierarchical structures and the <a href="https://github.com/YuLab-SMU/ecluster"  class="external-link" target="_blank" rel="noopener">ecluster</a> package (work in progress) to support various omics data structures provided in the Bioconductor project. This allows the related data at the feature or sample level to be interpreted and integrated according to their hierarchical relationships.</p>
</td>
</tr>
</table>
<!--

扩展树相关数据集成与可视化这一系列工具以应用于其它tree-like structures (e.g., hierarchical clustering and classification/regression trees) ，实现了ggtreeDendro用于支持通用的层次结构和ecluster包用于Bioconductor各种组学数据结构的支持。这使得feature或样本水平的相关数据能够基于层次结构进行解析和整合。
-->
<hr>
<h2 id="comments-from-the-academic-community">
  Comments from the academic community
  <a class="heading-link" href="#comments-from-the-academic-community">
    <i class="fa fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:75%">
<blockquote>
<p>It is embarrassing to me that I just learned about &ldquo;ggtree&rdquo; this month, but I am looking forward to incorporating it into several projects in my lab.  I am looking forward to using it to visualizing annotation results after doing BLAST or FASTA similarity searches, and for high lighting errors in protein sequences among closely related organisms.  I have been searching for a tool like &ldquo;ggtree&rdquo; for several years, not appreciating that you had already built it.  So my interest in recruiting your paper is somewhat selfish.  I would like more help in understanding how to add graphical annotations to the leaves of trees.</p>
<p>&ndash; The email (Sep 14, 2019) from Professor <a href="https://fasta.bioch.virginia.edu/wrpearson/"  class="external-link" target="_blank" rel="noopener">William Pearson</a> (ISCB Fellow) inviting me to contribute to <em>Current Protocols in Bioinformatics</em></p>
</blockquote>
</td>
<td style="border:none;">
<a href="https://fasta.bioch.virginia.edu/wrpearson/"><img src="https://fasta.bioch.virginia.edu/wrpearson/wrp_crop_2018.jpg" width='200px'/></a>
</td>
</tr>
</table>

  </article>
</section>

  

    </div>

    <footer class="footer">
  <section class="container">
    ©
    
      2018 -
    
    2023
     YuLab-SMU 
    ·
    
    Powered by <a href="https://gohugo.io/" target="_blank" rel="noopener">Hugo</a> & <a href="https://github.com/luizdepra/hugo-coder/" target="_blank" rel="noopener">Coder</a>.
    
  </section>
</footer>

  </main>

  

  
  
  <script src="/js/coder.min.6ae284be93d2d19dad1f02b0039508d9aab3180a12a06dcc71b0b0ef7825a317.js" integrity="sha256-auKEvpPS0Z2tHwKwA5UI2aqzGAoSoG3McbCw73gloxc="></script>
  

  

  


  

  

  

  <script data-goatcounter="https://code.goatcounter.com/count"
        async src="//gc.zgo.at/count.js"></script>


  
<script defer src='https://static.cloudflareinsights.com/beacon.min.js'
        data-cf-beacon='{"token": "token"}'></script>



  

  

  

  

  

  

  

  

  
</body>

</html>
