<!DOCTYPE html>
<html lang="en">

<head>
  <title>
  Biological Knowledge Mining and Discovery · YuLab@SMU
</title>
  <meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="color-scheme" content="light dark">




<meta name="author" content="YuLab-SMU">
<meta name="description" content="Knowledge discovery for precision medicine big data is an important aspect in promoting clinical translational applications. By mining biomedical knowledge, it is possible to drive the discovery of new biomedical knowledge. We have implemented a series of methods and tools, including: (1) for the first time, implementing biological theme comparison for complex experimental designs, (2) universal enrichment analysis methods for interpreting omics data, (3) measuring semantic similarity to assist in biological knowledge discovery, (4) cistromic data mining to aid in the discovery of (unknown) co-regulators, (5) integration of biological knowledge to enhance the biological interpretability of single-cell clustering, and (6) characterization of single-cell functional states and identification of spatial specific biological functions.">
<meta name="keywords" content="developer,bioinformatics,omics,visualization">

<meta name="twitter:card" content="summary"/><meta name="twitter:title" content="Biological Knowledge Mining and Discovery"/>
<meta name="twitter:description" content="Knowledge discovery for precision medicine big data is an important aspect in promoting clinical translational applications. By mining biomedical knowledge, it is possible to drive the discovery of new biomedical knowledge. We have implemented a series of methods and tools, including: (1) for the first time, implementing biological theme comparison for complex experimental designs, (2) universal enrichment analysis methods for interpreting omics data, (3) measuring semantic similarity to assist in biological knowledge discovery, (4) cistromic data mining to aid in the discovery of (unknown) co-regulators, (5) integration of biological knowledge to enhance the biological interpretability of single-cell clustering, and (6) characterization of single-cell functional states and identification of spatial specific biological functions."/>

<meta property="og:title" content="Biological Knowledge Mining and Discovery" />
<meta property="og:description" content="Knowledge discovery for precision medicine big data is an important aspect in promoting clinical translational applications. By mining biomedical knowledge, it is possible to drive the discovery of new biomedical knowledge. We have implemented a series of methods and tools, including: (1) for the first time, implementing biological theme comparison for complex experimental designs, (2) universal enrichment analysis methods for interpreting omics data, (3) measuring semantic similarity to assist in biological knowledge discovery, (4) cistromic data mining to aid in the discovery of (unknown) co-regulators, (5) integration of biological knowledge to enhance the biological interpretability of single-cell clustering, and (6) characterization of single-cell functional states and identification of spatial specific biological functions." />
<meta property="og:type" content="article" />
<meta property="og:url" content="http://yulab-smu.top/contribution-knowledge-mining/" /><meta property="article:section" content="" />








<link rel="canonical" href="http://yulab-smu.top/contribution-knowledge-mining/">


<link rel="preload" href="/fonts/forkawesome-webfont.woff2?v=1.2.0" as="font" type="font/woff2" crossorigin>


  
  
  <link rel="stylesheet" href="/css/coder.min.135e22c97ff685fe983fc60048e309ced8f00d8d38f536aa67dba8a13a03dfa4.css" integrity="sha256-E14iyX/2hf6YP8YASOMJztjwDY049TaqZ9uooToD36Q=" crossorigin="anonymous" media="screen" />








 




<link rel="icon" type="image/svg+xml" href="/images/favicon.svg" sizes="any">
<link rel="icon" type="image/png" href="/images/favicon-32x32.png" sizes="32x32">
<link rel="icon" type="image/png" href="/images/favicon-16x16.png" sizes="16x16">

<link rel="apple-touch-icon" href="/images/apple-touch-icon.png">
<link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon.png">

<link rel="manifest" href="/site.webmanifest">
<link rel="mask-icon" href="/images/safari-pinned-tab.svg" color="#5bbad5">









</head>




<body class="preload-transitions colorscheme-light">
  
<div class="float-container">
    <a id="dark-mode-toggle" class="colorscheme-toggle">
        <i class="fa fa-adjust fa-fw" aria-hidden="true"></i>
    </a>
</div>


  <main class="wrapper">
    <nav class="navigation">
  <section class="container">
    <a class="navigation-title" href="/">
      YuLab@SMU
    </a>
    
      <input type="checkbox" id="menu-toggle" />
      <label class="menu-button float-right" for="menu-toggle">
        <i class="fa fa-bars fa-fw" aria-hidden="true"></i>
      </label>
      <ul class="navigation-list">
        
          
            <li class="navigation-item">
              <a class="navigation-link" href="/about/">About</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/publications/">Publications</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/contributions/">Contributions</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/members/">Members</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/news/">News</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/contact/">Contact</a>
            </li>
          
        
        
      </ul>
    
  </section>
</nav>


    <div class="content">
      
  <section class="container page">
  <article>
    <header>
      <h1 class="title">
        <a class="title-link" href="http://yulab-smu.top/contribution-knowledge-mining/">
          Biological Knowledge Mining and Discovery
        </a>
      </h1>
    </header>

    <style>
    @media only screen and (max-width: 768px) {
        td {
            display: block;
        }
    }
</style>
<p>Knowledge discovery for precision medicine big data is an important aspect in promoting clinical translational applications. By mining biomedical knowledge, it is possible to drive the discovery of new biomedical knowledge. We have implemented a series of methods and tools, including: (1) for the first time, implementing biological theme comparison for complex experimental designs, (2) universal enrichment analysis methods for interpreting omics data, (3) measuring semantic similarity to assist in biological knowledge discovery, (4) cistromic data mining to aid in the discovery of (unknown) co-regulators, (5) integration of biological knowledge to enhance the biological interpretability of single-cell clustering, and (6) characterization of single-cell functional states and identification of spatial specific biological functions. This series of methods and software enables a more diverse range of biomedical knowledge to be applied at a greater variety of species, allowing biomedical knowledge to support biological big data mining and extract novel and potential disoveries.</p>
<!--
面向精准医学大数据的知识发现是促进临床转化应用的重要方面，通过挖掘生物医学知识能够驱动发现新的生物医学知识。我们实现了一系列方法和工具，包括（1）首次实现针对复杂实验设计比较生物学主题， （2）通用富集分析方法助力通过组学数据挖掘分子机制，（3）语义度量帮助生物学知识发现，（4）顺反子数据挖掘帮助发现（未知）共调控因子，（5）整合生物学知识提升单细胞聚类的生物学可解释性，（6）单细胞功能状态刻画和空间特异性的生物学功能鉴定。这一系列方法和软件使得更加多样化的生物医学知识能够服务应用于更多的物种，让生物医学知识助力生物大数据挖掘，提炼出新颖的和潜在的发现。
-->
<hr>
<h2 id="1-frist-time-implementation-of-comparing-biological-themes">
  1. Frist-time implementation of comparing biological themes
  <a class="heading-link" href="#1-frist-time-implementation-of-comparing-biological-themes">
    <i class="fa fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:25%">
<p><a href="http://online.liebertpub.com/doi/abs/10.1089/omi.2011.0118"><img src="/images/knowledge-mining/cp-cover.png" width='1000px'/></a></p>
</td>
<td style="border:none;">
<p>When experimental designs were essentially simple, such as a case-control design, we realized the importance of complex experimental designs involving multiple time points and conditions for studying biomedical issues. We devloped the <a href="https://www.bioconductor.org/packages/clusterProfiler"  class="external-link" target="_blank" rel="noopener">clusterProfiler</a> to compare biological themes, and this work was published in <a href="http://online.liebertpub.com/doi/abs/10.1089/omi.2011.0118"  class="external-link" target="_blank" rel="noopener"><em>OMICS: A Journal of Integrative Biology 2012</em></a>. It has been cited over 20,000 times, making it the most highly cited paper in the journal and one of the top 10 papers among China&rsquo;s highly cited papers from 2011 to 2021. Featuring an extremely user-friendly interface, it allows for specifying complex experimental designs using <a href="https://stat.ethz.ch/R-manual/R-devel/library/stats/html/formula.html"  class="external-link" target="_blank" rel="noopener">formula syntax</a>, as demonstrated in <a href="https://doi.org/10.1016/j.xinn.2021.100141"  class="external-link" target="_blank" rel="noopener"><em>The Innovation 2021</em></a> with data showcasing the different time points with various drug treatments simultaneously. Twelve years after its initial publication, we authored a protocol-type article introducing the functionality of comparing biological themes in comparing disease subtypes, functions of perturbed transcription factors, and cell type annotations using various omics data, including metagenomics, metabolomics, transcriptomics, and single-cell omics. This work was published in <a href="" ><em>Nature Protocols 2024</em></a>.</p>
</td>
</tr>
</table>
<!--
在实验设计基本上都是疾病-对照这样简单设计的时候，我们就意识到多时间点、多条件等复杂的实验设计对研究生物医学问题很重要，开发了clusterProfiler以比较生物学主题，这一工作发表于OMICS 2012。引用超过2万次，成为该期刊最高被引论文。支持使用formula形式来指定复杂的实验设计（如The Innovation 2021中展示的不同时间点和药物处理）。 时隔12年，我们撰写了protocol类文章，通过包括宏基因组、代谢组、转录组、单细胞组学在内的多种组学数据，介绍比较生物学主题这一功能在比较疾病亚型、受胁迫扰动转录因子功能和细胞类型注释中的应用，并发表在Nature Protocols 2024。

-->
<h2 id="2-discovering-potential-molecular-mechanisms-driven-by-high-throughput-data-and-biomedical-knowledge">
  2. Discovering potential molecular mechanisms driven by high-throughput data and biomedical knowledge
  <a class="heading-link" href="#2-discovering-potential-molecular-mechanisms-driven-by-high-throughput-data-and-biomedical-knowledge">
    <i class="fa fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:25%">
<p><a href="https://doi.org/10.1016/j.xinn.2021.100141"><img src="/images/knowledge-mining/graphic-abstract-The-Innovation-2021.jpg" width='1000px'/></a></p>
</td>
<td style="border:none;">
<p>One of the key aspects of functional genomics research is to identify the biological pathways involved in perturbed genes and propose hypotheses about molecular mechanisms. Gene enrichment analysis is a widely used and effective method. However, it faces challenges such as outdated annotation and lack of support for non-model organisms. We have developed a series of R packages represented by <a href="https://www.bioconductor.org/packages/clusterProfiler"  class="external-link" target="_blank" rel="noopener">clusterProfiler</a>  to address these challenges. ClusterProfiler allows for online retrieval of the latest genome annotations, supporting GO and KEGG enrichment analysis for thousands of species; providing a universal interface to support user-provided custom annotations, facilitating analysis of new species and using new functional annotations; allow for the use of genomic regions, enabling enrichment analysis of epigenomic data; <a href="/contribution-knowledge-mining/#1-frist-time-implementation-of-comparing-biological-themes" >implement comparative analysis of multiple datasets for supporting complex experimental designs</a>; and provide a tidy interface, making it easier for users to operate, explore and interpret data (<a href="http://online.liebertpub.com/doi/abs/10.1089/omi.2011.0118"  class="external-link" target="_blank" rel="noopener"><em>OMICS: A Journal of Integrative Biology 2012</em></a>; <a href="https://doi.org/10.1016/j.xinn.2021.100141"  class="external-link" target="_blank" rel="noopener"><em>The Innovation 2021</em></a>; <a href="" ><em>Nature Protocols 2024</em></a>).</p>
<p>This work has had a significant impact on biomedical research, with over 20,000 citations and integration into more than 40 bioinformatics software tools. It has become one of the indispensable bioinformatics tools for various omics data analyses. We have also extended its application to disease ontology (<a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btu684"  class="external-link" target="_blank" rel="noopener"><em>Bioinformatics 2015b</em></a>), medical subject headings (<a href="https://doi.org/10.1093/bioinformatics/bty410"  class="external-link" target="_blank" rel="noopener"><em>Bioinformatics 2018</em></a>), and Reactome pathway analysis (<a href="http://pubs.rsc.org/en/Content/ArticleLanding/2016/MB/C5MB00663E"  class="external-link" target="_blank" rel="noopener"><em>Molecular BioSystems 2016</em></a>).</p>
</td>
</tr>
</table>
<!--
如何定位到受扰动的基因所涉及的生物学通路，提出分子机制假说，是功能基因组研究的重要内容之一。基因富集分析是被广泛使用且行之有效的方法，然而存在知识库老旧、缺乏非模式生物支持和知识库单一等问题。我们开发了以clusterProfiler为代表的系列R包，通过在线获取最新的全基因组注释，支持几千个物种的GO和KEGG富集分析；提供通用方法支持用户提供的自定义知识库，便于分析新物种和使用新的功能注释；允许使用基因组位置信息，从而支持了表观基因组数据的富集分析；首次实现了对比分析多组数据的功能，以支持复杂的实验设计；并支持整洁接口，便于用户操作、探索和解读数据。

这一工作对生物医学研究产生了较大的影响，被整合于超过40个同行软件中，成为各种组学数据分析中不可或缺的生物信息学基础工具之一，我们也将其扩展应用于疾病本体学、医学主题词和Reactome通路的分析。
-->
<h2 id="3-measuring-gene-semantic-similarity-promotes-the-discovery-of-biomedical-knowledge">
  3. Measuring gene semantic similarity promotes the discovery of biomedical knowledge
  <a class="heading-link" href="#3-measuring-gene-semantic-similarity-promotes-the-discovery-of-biomedical-knowledge">
    <i class="fa fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:25%">
<p><a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/26/7/976"><img src="/images/knowledge-mining/semsim.png" width='1000px'/></a></p>
</td>
<td style="border:none;">
<p>By mining biomedical knowledge, new biomedical knowledge can be discovered. Calculating gene functional similarity using biomedical knowledge plays a crucial role in this process. It quantifies biomedical knowledge mathematically to measure the similarity between geens. Similarity entities share similar functions and behaviors, which form the basis for solving a wide range of real-world problems, including predicting and inferring aspects such as gene function, localization, and interactions, as well as analyzing diseases and drugs (including drug repurposing).</p>
<p>We developed the software tool <a href="https://www.bioconductor.org/packages/GOSemSim"  class="external-link" target="_blank" rel="noopener">GOSemSim</a> based on gene ontology, which implements multiple information content-based algorithms and graph-based algorithms. This work was published in <a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/26/7/976"  class="external-link" target="_blank" rel="noopener"><em>Bioinformatics 2010</em></a> and as a chapter in <a href="http://dx.doi.org/10.1007/978-1-0716-0301-7_11"  class="external-link" target="_blank" rel="noopener"><em>Stem Cell Transcriptional Networks (2nd edition) 2020</em></a>. We extended this work to a broader range of biomedical knowledge, supporting the measurement of gene similarity from perspectives such as diseases (<a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btu684"  class="external-link" target="_blank" rel="noopener"><em>Bioinformatics 2015b</em></a>), phenotypes, and medical subject headings (<a href="https://doi.org/10.1093/bioinformatics/bty410"  class="external-link" target="_blank" rel="noopener"><em>Bioinformatics 2018</em></a>).</p>
</td>
</tr>
</table>
<!--
通过挖掘生物医学知识能够驱动发现新的生物医学知识。其中利用生物医学知识计算基因的功能相似性起着至关重要的作用，它通过数学方式量化生物医学知识，以度量基因之间的相似性。相似的东西拥有相似的功能和行为，这是解决大量现实问题的基础，包括基因的功能、定位和相互作用等方面的预测和推断，以及疾病和药物的分析（包括药物功能重定位）等。我们基于基因本体学知识开发了软件工具GOSemSim，实现了多个基于信息含量的算法和基于图结构的算法，该工作发表于Bioinformatics 2010, 26(7):976-978，并受到主编Benjamin Kidder教授的邀请，在《Stem Cell Transcriptional Networks》（第二版）一书中撰写一章介绍这一工作（Methods in Molecular Biology 2020, 2117:207-215）。申请人将这一工作扩展到更为广泛的生物医学知识，支持基于疾病、表型和医学主题词等生物医学知识度量基因的相似性。
-->
<h2 id="4-annotation-visualization-data-mining-of-cistrome-data">
  4. Annotation, visualization, data mining of cistrome data
  <a class="heading-link" href="#4-annotation-visualization-data-mining-of-cistrome-data">
    <i class="fa fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:25%">
<p><a href="https://academic.oup.com/bioinformatics/article/31/14/2382/255379"><img src="/images/knowledge-mining/chipseeker.png" width='1000px'/></a></p>
</td>
<td style="border:none;">
<p>Cistrome refers to the set of cis-acting targets of trans-acting factors at the genome-wide scale, including transcription factor binding sites and the genomic locations of histone modifications. The cis-regulatory information can be obtained at the genome-wide level through techniques such as ChIP-seq, DNase-seq, and ATAC-seq. The key to bridging upstream (determining locations) and downstream analysis (functional studies) lies in annotating the genomic positional information. On the other hand, most tools are designed for individual datasets. With the proliferation of cistromic technologies and the accumulation of public data, comparing multiple datasets, mining public data, and identifying (unknown) co-regulators of transcription factors are critical for comprehensive understanding of cistrome interactions. To address these issues, we developed the <a href="https://www.bioconductor.org/packages/ChIPseeker"  class="external-link" target="_blank" rel="noopener">ChIPseeker</a> package for annotation and comparision, integrated with the GEO database to allow users to infer potential co-transcription factors or protein complexes through data mining. This work was published in <a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btv145"  class="external-link" target="_blank" rel="noopener">Bioinformatics 2015a</a>. ChIPseeker has been widely applied to various cistromic datasets and is listed as a key step in the anlysis of <a href="https://doi.org/10.1093/bib/bby057"  class="external-link" target="_blank" rel="noopener">DNase-seq</a> and <a href="https://informatics.fas.harvard.edu/atac-seq-guidelines.html"  class="external-link" target="_blank" rel="noopener">ATAC-seq</a> data. We also published a protocol article (<a href="https://doi.org/10.1002/cpz1.585"  class="external-link" target="_blank" rel="noopener"><em>Current Protocols 2022</em></a>) introducing the analysis application of ChIPseeker in various different epigenomic datasets.</p>
</td>
</tr>
</table>
<!--
顺反组（cistrome）指反式作用因子在全基因组尺度下的顺式作用靶点集合，包括转录作用因子结合位点和组蛋白修饰在全基因组上的位置。顺式调控信息可由ChIP-seq、DNase-seq和ATAC-seq在全基因组水平上获取，衔接上游（确定位置）和下游分析（功能研究）的重点在于对基因组位置信息进行注释。另一方面，大多数的工具针对单一的数据集而设计，随着顺反组技术的普及和公共数据的积累，比较多组数据、挖掘公共数据，鉴定反式作用因子的（未知）共同调控因子，是全面理解顺反子相互作用的关键。针对这些问题，我们开发了ChIPseeker包，进行注释和比较，并整合了GEO数据库，以允许用户通过数据挖掘，推断出潜在的共转录因子或蛋白复合体。该工作发表于Bioinformatics 2015a。ChIPseeker被广泛应用于各种顺反组数据，被列为DNase-seq和ATAC-seq数据分析指南的关键步骤。我们也撰写了protocol文章介绍于ChIPseeker在各种不同的表观基因组学数据中的分析应用。
-->
<h2 id="5-integrating-biological-knowledge-to-enhance-the-biological-interpretability-of-single-cell-clustering">
  5. Integrating biological knowledge to enhance the biological interpretability of single-cell clustering
  <a class="heading-link" href="#5-integrating-biological-knowledge-to-enhance-the-biological-interpretability-of-single-cell-clustering">
    <i class="fa fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:25%">
<p><a href="https://github.com/YuLab-SMU/MSGNN/"><img src="/images/knowledge-mining/msgnn.png" width='1000px'/></a></p>
</td>
<td style="border:none;">
<p>Clustering is a key step in bridging upstream and downstream analyses in single-cell analysis, and accurate clustering is crucial for downstream analysis. Currently, the most common clustering methods rely on graph-based community detection approaches. We propose integrating biological knowledge as attributes of graph nodes to obtain clustering results that are more aligned with biological interpretations. We have implemented the <a href="https://github.com/YuLab-SMU/MSGNN/"  class="external-link" target="_blank" rel="noopener">MSGNN</a> tool based on graph neural networks to enhance the biological interpretation of complex data.</p>
</td>
</tr>
</table>
<!--
聚类是衔接单细胞上下游分析的关键步骤，准确聚类对下游分析至关重要。当前最常见的聚类方法都是基于图的社区发现方法，我们提出将生物学知识作为网络图节点属性进行融合，从而可以得到更符合生物学解释的聚类结果。我们基于图神经网络实现了MSGNN工具，以增强对复杂数据的生物学解读。
-->
<h2 id="6-identification-of-spatial-variable-biological-functions">
  6. Identification of spatial variable biological functions
  <a class="heading-link" href="#6-identification-of-spatial-variable-biological-functions">
    <i class="fa fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:25%">
<p><a href="https://github.com/YuLab-SMU/SVP/"><img src="/images/knowledge-mining/svp.png" width='1000px'/></a></p>
</td>
<td style="border:none;">
<p>Identification of biological functions with spatial specific distributions involves identifying genes with high spatial variability, followed by enrichment analysis to characterize these functions. We propose a new method for characterizing cell functional states as well as a general approach for identifying spatial highly variable features. The <a href="https://github.com/YuLab-SMU/SVP/"  class="external-link" target="_blank" rel="noopener">SVP</a> package has been developed to implement these two methods, enabling the characterization of various biological functions, including biological pathways, and identifying whether these functions exhibit spatial distribution specificity.</p>
</td>
</tr>
</table>
<!--
当前对具有空间特异性分布的生物学功能的鉴定，主流方法是通过鉴定空间高变异性的基因，再衔接富集分析进行生物学功能的刻画。我们提出了通过单细胞转录组数据进行细胞功能状态的刻画方法，以及通用的空间高可变特征鉴定方法。开发了SVP包实现这两个方法，可以对包括生物学通路在内的各种生物学功能进行状态刻画，再鉴定这些功能是否具有空间分布的特异性。
-->
<h2 id="comments-from-the-academic-community">
  Comments from the academic community
  <a class="heading-link" href="#comments-from-the-academic-community">
    <i class="fa fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<blockquote>
<p>Users familiar with R/Bioconductor have several R packages to perform epigenomic enrichment analysis. <strong>ChIPseeker</strong> (Yu et al., 2015) provides a variety of functions for annotating user-provided ROIs in BED format by proximity to the nearby genes, finding overlaps with epigenomic regions, performing gene-centric and epigenomic enrichment analysis. <strong>ChIPseeker</strong> employs permutation strategy to calculate overlap enrichment p-values. <strong>ChIPseeker</strong> provides access to multi-organism epigenomic datasets from GEO (Gene Expression Omnibus), expanding enrichment analysis to species other than human. Coupled with excellent visualization capabilities, <strong>ChIPseeker</strong> is a well-designed tool with complete functionality for the interpretation of genome-wide ROIs.</p>
<p>&ndash; Excerpt from the article: <a href="https://academic.oup.com/bioinformatics/article/33/20/3323/3892391"  class="external-link" target="_blank" rel="noopener">Epigenomic annotation-based interpretation of genomic data: from enrichment analysis to machine learning</a>, <strong><em>Bioinformatics</em></strong>, 2017, 33(20):3323-3330.</p>
</blockquote>
<h2 id="publications">
  Publications
  <a class="heading-link" href="#publications">
    <i class="fa fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<font size="3">
<ul>
<li>S Xu<sup>#</sup>, E Hu<sup>#</sup>, Y Cai<sup>#</sup>, Z Xie<sup>#</sup> , X Luo<sup>#</sup>, L Zhan, W Tang, Q Wang, B Liu, R Wang, W Xie, T Wu, L Xie, <strong>G Yu</strong><sup>*</sup>. <a href="" >Using clusterProfiler to characterise Multi-Omics Data</a>. <strong><em>Nature Protocols</em></strong>. 2024, accepted.</li>
<li>Q Wang<sup>#</sup>, M Li<sup>#</sup>, T Wu, L Zhan, L Li, M Chen, W Xie, Z Xie, E Hu, S Xu, <strong>G Yu</strong><sup>*</sup>. <a href="https://doi.org/10.1002/cpz1.585"  class="external-link" target="_blank" rel="noopener">Exploring epigenomic datasets by ChIPseeker</a>. <strong><em>Current Protocols</em></strong>, 2022, 2(10): e585.</li>
<li>N Sato, Y Tamada, <strong>G Yu</strong>, Y Okuno<sup>*</sup>. <a href="https://doi.org/10.1093/bioinformatics/btac175"  class="external-link" target="_blank" rel="noopener">CBNplot: Bayesian network plots for enrichment analysis</a>. <strong><em>Bioinformatics</em></strong>. 2022, 38(10):2959-2960.</li>
<li>T Wu<sup>#</sup>, E Hu<sup>#</sup>, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo<sup>*</sup>, <strong>G Yu</strong><sup>*</sup>. <a href="https://doi.org/10.1016/j.xinn.2021.100141"  class="external-link" target="_blank" rel="noopener">clusterProfiler 4.0: A universal enrichment tool for interpreting omics data</a>. <strong><em>The Innovation</em></strong>. 2021, 2(3):100141.</li>
<li><strong>Yu G</strong>. <a href="http://dx.doi.org/10.1007/978-1-0716-0301-7_11"  class="external-link" target="_blank" rel="noopener">Gene Ontology Semantic Similarity Analysis Using GOSemSim</a>. In: Kidder B. (eds) Stem Cell Transcriptional Networks. <em><strong>Methods in Molecular Biology</strong></em>, 2020, 2117:207-215. Humana, New York, NY.</li>
<li>Z Hao, D Lv, Y Ge, J Shi, D Weijers, <strong>G Yu</strong><sup>*</sup>, J Chen<sup>*</sup>. <a href="https://peerj.com/articles/cs-251/"  class="external-link" target="_blank" rel="noopener">RIdeogram: drawing SVG graphics to visualize and map genome-wide data on the idiograms</a>. <strong><em>PeerJ Computer Science</em></strong>. 2020, 6:e251.</li>
<li><strong>G Yu</strong>. <a href="https://doi.org/10.1093/bioinformatics/bty410"  class="external-link" target="_blank" rel="noopener">Using meshes for MeSH term enrichment and semantic analyses</a>. <strong><em>Bioinformatics</em></strong>. 2018, 34(21):3766-3767.</li>
<li><strong>G Yu</strong>, QY He<sup>*</sup>. <a href="http://pubs.rsc.org/en/Content/ArticleLanding/2016/MB/C5MB00663E"  class="external-link" target="_blank" rel="noopener">ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</a>. <strong><em>Molecular BioSystems</em></strong>. 2016, 12(2):477-479.</li>
<li><strong>G Yu</strong><sup>*</sup>, LG Wang, QY He<sup>*</sup>. <a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btv145"  class="external-link" target="_blank" rel="noopener">ChIPseeker: an R/Bioconductor package for ChIP peak annotation, comparision and visualization</a>. <strong><em>Bioinformatics</em></strong>. 2015, 31(14):2382-2383.</li>
<li><strong>G Yu</strong><sup>*</sup>, LG Wang, GR Yan, QY He<sup>*</sup>. <a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btu684"  class="external-link" target="_blank" rel="noopener">DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis</a>. <strong><em>Bioinformatics</em></strong>. 2015, 31(4):608-609.</li>
<li><strong>G Yu</strong>, LG Wang, Y Han, QY He<sup>*</sup>. <a href="http://online.liebertpub.com/doi/abs/10.1089/omi.2011.0118"  class="external-link" target="_blank" rel="noopener">clusterProfiler: an R package for comparing biological themes among gene clusters</a>. <strong><em>OMICS: A Journal of Integrative Biology</em></strong>. 2012, 16(5):284-287.</li>
<li><strong>G Yu</strong>, QY He<sup>*</sup>. <a href="https://jclinbioinformatics.biomedcentral.com/articles/10.1186/2043-9113-1-15"  class="external-link" target="_blank" rel="noopener">Functional similarity analysis of human virus-encoded miRNAs</a>. <strong><em>Journal of Clinical Bioinformatics</em></strong>, 2011, 1(1):15.</li>
<li><strong>G Yu</strong><sup>#</sup>, CL Xiao<sup>#</sup>, X Bo, CH Lu, Y Qin, S Zhan, QY He<sup>*</sup>. <a href="http://www.jiomics.com/index.php/jio/article/view/21"  class="external-link" target="_blank" rel="noopener">A new method for measuring functional similarity of microRNAs</a>. <strong><em>Journal of Integrated OMICS</em></strong>, 2011, 1(1):49-54.</li>
<li><strong>G Yu</strong><sup>#</sup>, F Li<sup>#</sup>, Y Qin, X Bo<sup>*</sup>, Y Wu, S Wang<sup>*</sup>. <a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/26/7/976"  class="external-link" target="_blank" rel="noopener">GOSemSim: an R package for measuring semantic similarity among GO terms and gene products</a>. <strong><em>Bioinformatics</em></strong>. 2010, 26(7):976-978.</li>
</ul>
</font>

  </article>
</section>

  

    </div>

    <footer class="footer">
  <section class="container">
    ©
    
      2018 -
    
    2024
     YuLab-SMU 
    ·
    
    Powered by <a href="https://gohugo.io/" target="_blank" rel="noopener">Hugo</a> & <a href="https://github.com/luizdepra/hugo-coder/" target="_blank" rel="noopener">Coder</a>.
    
  </section>
</footer>

  </main>

  

  
  
  <script src="/js/coder.min.6ae284be93d2d19dad1f02b0039508d9aab3180a12a06dcc71b0b0ef7825a317.js" integrity="sha256-auKEvpPS0Z2tHwKwA5UI2aqzGAoSoG3McbCw73gloxc="></script>
  

  

  


  

  

  

  <script data-goatcounter="https://code.goatcounter.com/count"
        async src="//gc.zgo.at/count.js"></script>


  
<script defer src='https://static.cloudflareinsights.com/beacon.min.js'
        data-cf-beacon='{"token": "token"}'></script>



  

  

  

  

  

  

  

  

  
</body>

</html>
