<!DOCTYPE html>
<html lang="en">

<head>
  <title>
  Semantic Knowledge Mining: Methods and Applications · YuLab@SMU
</title>
  <meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="color-scheme" content="light dark">




<meta name="author" content="YuLab-SMU">
<meta name="description" content="
We have developed methods and tools for biological knowledge mining, including: (1) semantic similarity measurement; (2) universal enrichment analysis; (3) comparative analysis of biological themes; (4) cistrome data analysis; (5) single-cell data analysis; and (6) spatial transcriptomics analysis. These methods and software tools support the application of biomedical knowledge across diverse species, facilitating the mining of biological data and the discovery of novel insights.">
<meta name="keywords" content="developer,bioinformatics,omics,visualization">


  <meta name="twitter:card" content="summary">
  <meta name="twitter:title" content="Semantic Knowledge Mining: Methods and Applications">
  <meta name="twitter:description" content="We have developed methods and tools for biological knowledge mining, including: (1) semantic similarity measurement; (2) universal enrichment analysis; (3) comparative analysis of biological themes; (4) cistrome data analysis; (5) single-cell data analysis; and (6) spatial transcriptomics analysis. These methods and software tools support the application of biomedical knowledge across diverse species, facilitating the mining of biological data and the discovery of novel insights.">

<meta property="og:url" content="http://yulab-smu.top/contribution-knowledge-mining/">
  <meta property="og:site_name" content="YuLab@SMU">
  <meta property="og:title" content="Semantic Knowledge Mining: Methods and Applications">
  <meta property="og:description" content="We have developed methods and tools for biological knowledge mining, including: (1) semantic similarity measurement; (2) universal enrichment analysis; (3) comparative analysis of biological themes; (4) cistrome data analysis; (5) single-cell data analysis; and (6) spatial transcriptomics analysis. These methods and software tools support the application of biomedical knowledge across diverse species, facilitating the mining of biological data and the discovery of novel insights.">
  <meta property="og:locale" content="en">
  <meta property="og:type" content="article">




<link rel="canonical" href="http://yulab-smu.top/contribution-knowledge-mining/">


<link rel="preload" href="/fonts/fa-brands-400.woff2" as="font" type="font/woff2" crossorigin>
<link rel="preload" href="/fonts/fa-regular-400.woff2" as="font" type="font/woff2" crossorigin>
<link rel="preload" href="/fonts/fa-solid-900.woff2" as="font" type="font/woff2" crossorigin>


  
  
  <link rel="stylesheet" href="/css/coder.min.0ab58f0bc42b4f21bff7ac349245f3c5579ad3c46dee51b49ad860d174431d73.css" integrity="sha256-CrWPC8QrTyG/96w0kkXzxVea08Rt7lG0mthg0XRDHXM=" crossorigin="anonymous" media="screen" />








 




<link rel="icon" type="image/svg+xml" href="/images/favicon.svg" sizes="any">
<link rel="icon" type="image/png" href="/images/favicon-32x32.png" sizes="32x32">
<link rel="icon" type="image/png" href="/images/favicon-16x16.png" sizes="16x16">

<link rel="apple-touch-icon" href="/images/apple-touch-icon.png">
<link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon.png">

<link rel="manifest" href="/site.webmanifest">
<link rel="mask-icon" href="/images/safari-pinned-tab.svg" color="#5bbad5">









</head>




<body class="preload-transitions colorscheme-light">
  
<div class="float-container">
    <a id="dark-mode-toggle" class="colorscheme-toggle">
        <i class="fa-solid fa-adjust fa-fw" aria-hidden="true"></i>
    </a>
</div>


  <main class="wrapper">
    <nav class="navigation">
  <section class="container">
    
    <a class="navigation-title" href="http://yulab-smu.top/">
      YuLab@SMU
    </a>
    
    
      <input type="checkbox" id="menu-toggle" />
      <label class="menu-button float-right" for="menu-toggle">
        <i class="fa-solid fa-bars fa-fw" aria-hidden="true"></i>
      </label>
      <ul class="navigation-list">
        
          
            <li class="navigation-item">
              <a class="navigation-link " href="/about/">About</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link " href="/publications/">Publications</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link " href="/books/">Books</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link " href="/contributions/">Contributions</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link " href="/members/">Members</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link " href="/news/">News</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link " href="/contact/">Contact</a>
            </li>
          
        
        
      </ul>
    
  </section>
</nav>


    <div class="content">
      
  <section class="container page">
  <article>
    <header>
      <h1 class="title">
        <a class="title-link" href="http://yulab-smu.top/contribution-knowledge-mining/">
          Semantic Knowledge Mining: Methods and Applications
        </a>
      </h1>
    </header>

    <style>
    @media only screen and (max-width: 768px) {
        td {
            display: block;
        }
    }
</style>
<p>We have developed methods and tools for biological knowledge mining, including: (1) semantic similarity measurement; (2) universal enrichment analysis; (3) comparative analysis of biological themes; (4) cistrome data analysis; (5) single-cell data analysis; and (6) spatial transcriptomics analysis. These methods and software tools support the application of biomedical knowledge across diverse species, facilitating the mining of biological data and the discovery of novel insights.</p>
<hr>
<h2 id="1-semantic-similarity-measurement">
  1. Semantic Similarity Measurement
  <a class="heading-link" href="#1-semantic-similarity-measurement">
    <i class="fa-solid fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:25%">
<p><a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/26/7/976"><img src="/images/knowledge-mining/semsim.png" style="width:100%; max-width:1000px;"/></a></p>
</td>
<td style="border:none;">
<p>Gene functional similarity calculations mathematically quantify biomedical knowledge to gauge the similarity between genes. This forms the basis for solving various challenges, such as predicting gene attributes and analyzing diseases and drugs. To facilitate this process, we developed the software tool <a href="https://www.bioconductor.org/packages/GOSemSim"  class="external-link" target="_blank" rel="noopener">GOSemSim</a>, built upon gene ontology principles. This work was published in <a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/26/7/976"  class="external-link" target="_blank" rel="noopener"><em>Bioinformatics</em></a> in 2010. We have extended this work to encompass diverse biomedical knowledge domains, including diseases, phenotypes, and medical subject headings.</p>
</td>
</tr>
</table>
<h2 id="2-functional-enrichment-analysis">
  2. Functional Enrichment Analysis
  <a class="heading-link" href="#2-functional-enrichment-analysis">
    <i class="fa-solid fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:25%">
<p><a href="https://doi.org/10.1016/j.xinn.2021.100141"><img src="/images/knowledge-mining/graphic-abstract-The-Innovation-2021.jpg" style="width:100%; max-width:1000px;"/></a></p>
</td>
<td style="border:none;">
<p>To identify biological pathways associated with perturbed genes, we developed a suite of R packages, spearheaded by <a href="https://www.bioconductor.org/packages/clusterProfiler"  class="external-link" target="_blank" rel="noopener">clusterProfiler</a>. This tool facilitates online retrieval of genome annotations, supporting enrichment analysis for thousands of species. It also provides a universal interface for user-provided custom annotations. Notably, <a href="https://www.bioconductor.org/packages/clusterProfiler"  class="external-link" target="_blank" rel="noopener">clusterProfiler</a> supports the utilization of genomic regions for enrichment analysis. We have extended its application to disease ontology, medical subject headings, and Reactome pathway analysis. This work has been widely cited and integrated into numerous bioinformatics software tools.</p>
</td>
</tr>
</table>
<h2 id="3-comparative-biological-theme-analysis">
  3. Comparative Biological Theme Analysis
  <a class="heading-link" href="#3-comparative-biological-theme-analysis">
    <i class="fa-solid fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:25%">
<p><a href="http://online.liebertpub.com/doi/abs/10.1089/omi.2011.0118"><img src="/images/knowledge-mining/cp-cover.png" style="width:100%; max-width:1000px;"/></a></p>
</td>
<td style="border:none;">
<p>To address complex experimental designs involving multiple time points and conditions, we developed <a href="https://www.bioconductor.org/packages/clusterProfiler"  class="external-link" target="_blank" rel="noopener">clusterProfiler</a> for comparing biological themes. This contribution was published in <a href="http://online.liebertpub.com/doi/abs/10.1089/omi.2011.0118"  class="external-link" target="_blank" rel="noopener"><em>OMICS: A Journal of Integrative Biology</em></a> in 2012. The tool enables users to specify complex experimental designs using formula syntax, allowing for the comparison of functional profiles across different conditions. We authored a protocol article in <a href="https://www.nature.com/articles/s41596-024-01020-z"  class="external-link" target="_blank" rel="noopener"><em>Nature Protocols</em></a> in 2024, detailing the application of this comparative framework.</p>
</td>
</tr>
</table>
<h2 id="4-cistrome-data-analysis">
  4. Cistrome Data Analysis
  <a class="heading-link" href="#4-cistrome-data-analysis">
    <i class="fa-solid fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:25%">
<p><a href="https://academic.oup.com/bioinformatics/article/31/14/2382/255379"><img src="/images/knowledge-mining/chipseeker.png" style="width:100%; max-width:1000px;"/></a></p>
</td>
<td style="border:none;">
<p>We developed the <a href="https://www.bioconductor.org/packages/ChIPseeker"  class="external-link" target="_blank" rel="noopener">ChIPseeker</a> package to bridge the gap between genomic cis-regulatory positions and biological function. It facilitates annotation and comparison of genomic peaks, integrating with the GEO database for data mining. This work was published in <a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btv145"  class="external-link" target="_blank" rel="noopener">Bioinformatics</a> in 2015. ChIPseeker is widely used for analyzing various cistromic datasets, including ChIP-seq, DNase-seq, and ATAC-seq.</p>
</td>
</tr>
</table>
<h2 id="5-single-cell-data-analysis">
  5. Single-Cell Data Analysis
  <a class="heading-link" href="#5-single-cell-data-analysis">
    <i class="fa-solid fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:25%">
<p><a href="https://github.com/YuLab-SMU/MSGNN/"><img src="/images/knowledge-mining/msgnn.png" style="width:100%; max-width:1000px;"/></a></p>
</td>
<td style="border:none;">
<p>To enhance the interpretability of single-cell clustering, we proposed integrating biological knowledge as attributes of graph nodes. We developed the <a href="https://github.com/YuLab-SMU/MSGNN/"  class="external-link" target="_blank" rel="noopener">MSGNN</a> tool based on graph neural networks to realize this advancement, improving the biological interpretation of complex data.</p>
</td>
</tr>
</table>
<h2 id="6-spatial-transcriptomics-analysis">
  6. Spatial Transcriptomics Analysis
  <a class="heading-link" href="#6-spatial-transcriptomics-analysis">
    <i class="fa-solid fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:25%">
<p><a href="https://github.com/YuLab-SMU/SVP/"><img src="/images/knowledge-mining/svp.png" style="width:100%; max-width:1000px;"/></a></p>
</td>
<td style="border:none;">
<p>We proposed a method for characterizing cell functional states and identifying spatially highly variable features. We developed the <a href="https://github.com/YuLab-SMU/SVP/"  class="external-link" target="_blank" rel="noopener">SVP</a> package to facilitate these methods, enabling the characterization of biological functions and the determination of their spatial distribution specificity.</p>
</td>
</tr>
</table>
<h2 id="feedback-from-the-academic-community">
  Feedback from the academic community
  <a class="heading-link" href="#feedback-from-the-academic-community">
    <i class="fa-solid fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<table style="border:none; font-size: 90%; width:100%;">
<tr style="border:none;"> 
<td style="border:none;width:75%">
<blockquote>
<p>Users familiar with R/Bioconductor have several R packages to perform epigenomic enrichment analysis. <strong>ChIPseeker</strong> (Yu et al., 2015) provides a variety of functions for annotating user-provided ROIs in BED format by proximity to the nearby genes, finding overlaps with epigenomic regions, performing gene-centric and epigenomic enrichment analysis. <strong>ChIPseeker</strong> employs permutation strategy to calculate overlap enrichment p-values. <strong>ChIPseeker</strong> provides access to multi-organism epigenomic datasets from GEO (Gene Expression Omnibus), expanding enrichment analysis to species other than human. Coupled with excellent visualization capabilities, <strong>ChIPseeker</strong> is a well-designed tool with complete functionality for the interpretation of genome-wide ROIs.</p>
<p>&ndash; Excerpt from the article: <a href="https://academic.oup.com/bioinformatics/article/33/20/3323/3892391"  class="external-link" target="_blank" rel="noopener">Epigenomic annotation-based interpretation of genomic data: from enrichment analysis to machine learning</a>, <strong><em>Bioinformatics</em></strong>, 2017, 33(20):3323-3330.</p>
</blockquote>
</td>
</tr>
</table>
<h2 id="publications">
  Publications
  <a class="heading-link" href="#publications">
    <i class="fa-solid fa-link" aria-hidden="true" title="Link to heading"></i>
    <span class="sr-only">Link to heading</span>
  </a>
</h2>
<font size="3">
<ul>
<li><strong>G Yu</strong><sup>*</sup>. <a href="https://doi.org/10.59717/j.xinn-life.2025.100181"  class="external-link" target="_blank" rel="noopener">Background bias in functional enrichment analysis: Insights from clusterProfiler</a>. <strong><em>The Innovation Life</em></strong>. 2026, 4(1):100181.</li>
<li><strong>G Yu</strong><sup>*</sup>. <a href="https://doi.org/10.1016/j.xinn.2024.100722"  class="external-link" target="_blank" rel="noopener">Thirteen years of clusterProfiler</a>. <strong><em>The Innovation</em></strong>. 2024, 5(6):100722.</li>
<li>S Xu<sup>#</sup>, E Hu<sup>#</sup>, Y Cai<sup>#</sup>, Z Xie<sup>#</sup>, X Luo<sup>#</sup>, L Zhan, W Tang, Q Wang, B Liu, R Wang, W Xie, T Wu, L Xie, <strong>G Yu</strong><sup>*</sup>. <a href="https://www.nature.com/articles/s41596-024-01020-z"  class="external-link" target="_blank" rel="noopener">Using clusterProfiler to characterise Multi-Omics Data</a>. <strong><em>Nature Protocols</em></strong>. 2024, 19(11):3292-3320.</li>
<li>Q Wang<sup>#</sup>, M Li<sup>#</sup>, T Wu, L Zhan, L Li, M Chen, W Xie, Z Xie, E Hu, S Xu, <strong>G Yu</strong><sup>*</sup>. <a href="https://doi.org/10.1002/cpz1.585"  class="external-link" target="_blank" rel="noopener">Exploring epigenomic datasets by ChIPseeker</a>. <strong><em>Current Protocols</em></strong>, 2022, 2(10): e585.</li>
<li>N Sato, Y Tamada, <strong>G Yu</strong>, Y Okuno<sup>*</sup>. <a href="https://doi.org/10.1093/bioinformatics/btac175"  class="external-link" target="_blank" rel="noopener">CBNplot: Bayesian network plots for enrichment analysis</a>. <strong><em>Bioinformatics</em></strong>. 2022, 38(10):2959-2960.</li>
<li>T Wu<sup>#</sup>, E Hu<sup>#</sup>, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo<sup>*</sup>, <strong>G Yu</strong><sup>*</sup>. <a href="https://doi.org/10.1016/j.xinn.2021.100141"  class="external-link" target="_blank" rel="noopener">clusterProfiler 4.0: A universal enrichment tool for interpreting omics data</a>. <strong><em>The Innovation</em></strong>. 2021, 2(3):100141.</li>
<li><strong>Yu G</strong>. <a href="http://dx.doi.org/10.1007/978-1-0716-0301-7_11"  class="external-link" target="_blank" rel="noopener">Gene Ontology Semantic Similarity Analysis Using GOSemSim</a>. In: Kidder B. (eds) Stem Cell Transcriptional Networks. <em><strong>Methods in Molecular Biology</strong></em>, 2020, 2117:207-215. Humana, New York, NY.</li>
<li>Z Hao, D Lv, Y Ge, J Shi, D Weijers, <strong>G Yu</strong><sup>*</sup>, J Chen<sup>*</sup>. <a href="https://peerj.com/articles/cs-251/"  class="external-link" target="_blank" rel="noopener">RIdeogram: drawing SVG graphics to visualize and map genome-wide data on the idiograms</a>. <strong><em>PeerJ Computer Science</em></strong>. 2020, 6:e251.</li>
<li><strong>G Yu</strong>. <a href="https://doi.org/10.1093/bioinformatics/bty410"  class="external-link" target="_blank" rel="noopener">Using meshes for MeSH term enrichment and semantic analyses</a>. <strong><em>Bioinformatics</em></strong>. 2018, 34(21):3766-3767.</li>
<li><strong>G Yu</strong>, QY He<sup>*</sup>. <a href="http://pubs.rsc.org/en/Content/ArticleLanding/2016/MB/C5MB00663E"  class="external-link" target="_blank" rel="noopener">ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</a>. <strong><em>Molecular BioSystems</em></strong>. 2016, 12(2):477-479.</li>
<li><strong>G Yu</strong><sup>*</sup>, LG Wang, QY He<sup>*</sup>. <a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btv145"  class="external-link" target="_blank" rel="noopener">ChIPseeker: an R/Bioconductor package for ChIP peak annotation, comparision and visualization</a>. <strong><em>Bioinformatics</em></strong>. 2015, 31(14):2382-2383.</li>
<li><strong>G Yu</strong><sup>*</sup>, LG Wang, GR Yan, QY He<sup>*</sup>. <a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btu684"  class="external-link" target="_blank" rel="noopener">DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis</a>. <strong><em>Bioinformatics</em></strong>. 2015, 31(4):608-609.</li>
<li><strong>G Yu</strong>, LG Wang, Y Han, QY He<sup>*</sup>. <a href="http://online.liebertpub.com/doi/abs/10.1089/omi.2011.0118"  class="external-link" target="_blank" rel="noopener">clusterProfiler: an R package for comparing biological themes among gene clusters</a>. <strong><em>OMICS: A Journal of Integrative Biology</em></strong>. 2012, 16(5):284-287.</li>
<li><strong>G Yu</strong>, QY He<sup>*</sup>. <a href="https://jclinbioinformatics.biomedcentral.com/articles/10.1186/2043-9113-1-15"  class="external-link" target="_blank" rel="noopener">Functional similarity analysis of human virus-encoded miRNAs</a>. <strong><em>Journal of Clinical Bioinformatics</em></strong>, 2011, 1(1):15.</li>
<li><strong>G Yu</strong><sup>#</sup>, CL Xiao<sup>#</sup>, X Bo, CH Lu, Y Qin, S Zhan, QY He<sup>*</sup>. <a href="http://www.jiomics.com/index.php/jio/article/view/21"  class="external-link" target="_blank" rel="noopener">A new method for measuring functional similarity of microRNAs</a>. <strong><em>Journal of Integrated OMICS</em></strong>, 2011, 1(1):49-54.</li>
<li><strong>G Yu</strong><sup>#</sup>, F Li<sup>#</sup>, Y Qin, X Bo<sup>*</sup>, Y Wu, S Wang<sup>*</sup>. <a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/26/7/976"  class="external-link" target="_blank" rel="noopener">GOSemSim: an R package for measuring semantic similarity among GO terms and gene products</a>. <strong><em>Bioinformatics</em></strong>. 2010, 26(7):976-978.</li>
</ul>
</font>

  </article>
</section>

  

    </div>

    <footer class="footer">
  <section class="container">
    ©
    
      2018 -
    
    2026
     YuLab-SMU 
    ·
    
    Powered by <a href="https://gohugo.io/" target="_blank" rel="noopener">Hugo</a> & <a href="https://github.com/luizdepra/hugo-coder/" target="_blank" rel="noopener">Coder</a>.
    
  </section>
</footer>

  </main>

  

  
  
  <script src="/js/coder.min.6ae284be93d2d19dad1f02b0039508d9aab3180a12a06dcc71b0b0ef7825a317.js" integrity="sha256-auKEvpPS0Z2tHwKwA5UI2aqzGAoSoG3McbCw73gloxc="></script>
  

  

  


  

  

  

  <script data-goatcounter="https://code.goatcounter.com/count"
        async src="//gc.zgo.at/count.js"></script>


  
<script defer src='https://static.cloudflareinsights.com/beacon.min.js'
        data-cf-beacon='{"token": "token"}'></script>



  

  

  

  

  

  

  

  

  

  

  

</body>

</html>
