<!DOCTYPE html>
<html lang="en">

  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="Content-Language" content="en">

    <meta name="author" content="YuLab-SMU">
    <meta name="description" content="You can visit Google Scholar for details.
# Co-first authors; * Co-corresponding authors
 2022  N Sato, Y Tamada, G Yu, Y Okuno*. CBNplot: Bayesian network plots for enrichment analysis. Bioinformatics. 2022. T Wu, CL Xiao, TT Lam, G Yu*. The frontier status and trends in biomedical data visualization. Frontiers in Genetics. 2022. Altered gut microbiota as an auxiliary diagnostic indicator for patients with fracture-related infection. Frontiers in Microbiology. 2022.">
    <meta name="keywords" content="blog,developer,bioinformatics,omics,visualization">

    <meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="Publications"/>
<meta name="twitter:description" content="You can visit Google Scholar for details.
# Co-first authors; * Co-corresponding authors
 2022  N Sato, Y Tamada, G Yu, Y Okuno*. CBNplot: Bayesian network plots for enrichment analysis. Bioinformatics. 2022. T Wu, CL Xiao, TT Lam, G Yu*. The frontier status and trends in biomedical data visualization. Frontiers in Genetics. 2022. Altered gut microbiota as an auxiliary diagnostic indicator for patients with fracture-related infection. Frontiers in Microbiology. 2022."/>

    <meta property="og:title" content="Publications" />
<meta property="og:description" content="You can visit Google Scholar for details.
# Co-first authors; * Co-corresponding authors
 2022  N Sato, Y Tamada, G Yu, Y Okuno*. CBNplot: Bayesian network plots for enrichment analysis. Bioinformatics. 2022. T Wu, CL Xiao, TT Lam, G Yu*. The frontier status and trends in biomedical data visualization. Frontiers in Genetics. 2022. Altered gut microbiota as an auxiliary diagnostic indicator for patients with fracture-related infection. Frontiers in Microbiology. 2022." />
<meta property="og:type" content="article" />
<meta property="og:url" content="http://yulab-smu.top/publications/" /><meta property="article:section" content="" />





    <title>
  Publications Â· YuLab@SMU
</title>

    
      <link rel="canonical" href="http://yulab-smu.top/publications/">
    

    <link rel="preconnect" href="https://fonts.gstatic.com">
    <link href="https://fonts.googleapis.com/css?family=Lato:400,700%7CMerriweather:300,700%7CSource+Code+Pro:400,700&display=swap" rel="stylesheet">
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/fork-awesome@1.1.7/css/fork-awesome.min.css"
      integrity="sha256-gsmEoJAws/Kd3CjuOQzLie5Q3yshhvmo7YNtBG7aaEY=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/normalize.css@8/normalize.min.css">
    
      
      
      <link rel="stylesheet" href="/css/coder.min.f01c647a0d25b40da992a37c3376291185eed8a50ced8c26cc2c0bcfe38c97df.css" integrity="sha256-8Bxkeg0ltA2pkqN8M3YpEYXu2KUM7YwmzCwLz&#43;OMl98=" crossorigin="anonymous" media="screen" />
    

    

    

    

    

    <link rel="icon" type="image/png" href="/images/favicon-32x32.png" sizes="32x32">
    <link rel="icon" type="image/png" href="/images/favicon-16x16.png" sizes="16x16">

    <link rel="apple-touch-icon" href="/images/apple-touch-icon.png">
    <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon.png">

    

    <meta name="generator" content="Hugo 0.94.2" />
  </head>

  
  
  <body class="colorscheme-light"
        onload=""
  >
    
<div class="float-container">
    <a id="dark-mode-toggle" class="colorscheme-toggle">
        <i class="fa fa-adjust fa-fw" aria-hidden="true"></i>
    </a>
</div>


    <main class="wrapper">
      <nav class="navigation">
  <section class="container">
    <a class="navigation-title" href="/">
      YuLab@SMU
    </a>
    
      <input type="checkbox" id="menu-toggle" />
      <label class="menu-button float-right" for="menu-toggle">
        <i class="fa fa-bars fa-fw" aria-hidden="true"></i>
      </label>
      <ul class="navigation-list">
        
          
            <li class="navigation-item">
              <a class="navigation-link" href="/about/">About</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/publications/">Publications</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/projects/">Projects</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/software/">Software</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/members/">Members</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/documents/">Documents</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/posts/">Blog</a>
            </li>
          
            <li class="navigation-item">
              <a class="navigation-link" href="/contact/">Contact</a>
            </li>
          
        
        
      </ul>
    
  </section>
</nav>


      <div class="content">
        
  <section class="container page">
  <article>
    <header>
      <h1>Publications</h1>
    </header>

    <p>You can visit <a href="https://scholar.google.com/citations?user=DO5oG40AAAAJ&amp;hl=en">Google Scholar</a> for details.</p>
<p><sup>#</sup> Co-first authors; <sup>*</sup> Co-corresponding authors</p>
<h2 id="i-classfa-fa-calendar-fa-1x-aria-hiddentruei-2022"><i class="fa fa-calendar fa-1x" aria-hidden="true"></i> 2022</h2>
<ul>
<li>N Sato, Y Tamada, <strong>G Yu</strong>, Y Okuno<sup>*</sup>. <a href="https://doi.org/10.1093/bioinformatics/btac175">CBNplot: Bayesian network plots for enrichment analysis</a>. <strong><em>Bioinformatics</em></strong>. 2022.</li>
<li>T Wu, CL Xiao, TT Lam, <strong>G Yu</strong><sup>*</sup>. <a href="https://www.frontiersin.org/articles/10.3389/fgene.2022.890775/full">The frontier status and trends in biomedical data visualization</a>. <strong><em>Frontiers in Genetics</em></strong>. 2022.</li>
<li>Altered gut microbiota as an auxiliary diagnostic indicator for patients with fracture-related infection. <strong><em>Frontiers in Microbiology</em></strong>. 2022.</li>
</ul>
<h2 id="i-classfa-fa-calendar-fa-1x-aria-hiddentruei-2021"><i class="fa fa-calendar fa-1x" aria-hidden="true"></i> 2021</h2>
<ul>
<li>S Xu<sup>#</sup>, M Chen<sup>#</sup>, T Feng, L Zhan, L Zhou, <strong>G Yu</strong><sup>*</sup>. <a href="https://www.frontiersin.org/articles/10.3389/fgene.2021.774846">Use ggbreak to effectively utilize plotting space to deal with large datasets and outliers</a>. <strong><em>Frontiers in Genetics</em></strong>. 2021, 12:774846.</li>
<li>CH Gao, <strong>G Yu</strong>, P Cai<sup>*</sup>. <a href="https://doi.org/10.3389/fgene.2021.706907">ggVennDiagram: An intuitive, easy-to-use, and highly customizable R package to generate Venn diagram</a>. <strong><em>Frontiers in Genetics</em></strong>. 2021, 12:706907.</li>
<li>S Xu, Z Dai, P Guo, X Fu, S Liu, L Zhou, W Tang, T Feng, M Chen, L Zhan, T Wu, E Hu, Y Jiang<sup>*</sup>, X Bo<sup>*</sup>, <strong>G Yu</strong><sup>*</sup>. <a href="https://doi.org/10.1093/molbev/msab166">ggtreeExtra: Compact visualization of richly annotated phylogenetic data</a>. <strong><em>Molecular Biology and Evolution</em></strong>. 2021, 38(9):4039-4042.</li>
<li>T Wu<sup>#</sup>, E Hu<sup>#</sup>, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo<sup>*</sup>, <strong>G Yu</strong><sup>*</sup>. <a href="https://doi.org/10.1016/j.xinn.2021.100141">clusterProfiler 4.0: A universal enrichment tool for interpreting omics data</a>. <strong><em>The Innovation</em></strong>. 2021, 2(3):100141. (<strong><em>Feature article</em></strong> <sup id="fnref:1"><a href="#fn:1" class="footnote-ref" role="doc-noteref">1</a></sup>)</li>
<li>D Zeng<sup>#</sup>, Z Ye<sup>#</sup>, R Shen, <strong>G Yu</strong>, J Wu, Y Xiong, R Zhou, W Qiu, N Huang, L Sun, X Li, J Bin, Y Liao, M Shi, W Liao<sup>*</sup>. <a href="https://doi.org/10.3389/fimmu.2021.687975">IOBR: Multi-Omics Immuno-Oncology Biological Research to Decode Tumor Microenvironment and Signatures</a>. <strong><em>Frontiers in Immunology</em></strong>. 2021, 12:687975.</li>
<li>T Wu, E Hu, X Ge<sup>*</sup>, <strong>G Yu</strong><sup>*</sup>. <a href="https://doi.org/10.7717/peerj.11421">nCov2019: an R package for studying the COVID-19 coronavirus pandemic</a>. <strong><em>PeerJ</em></strong>. 2021, 9:e11421.</li>
<li>R Huang, C Soneson, FGM Ernst, KC Rue-Albrecht, <strong>G Yu</strong>, SC Hicks, MD Robinson<sup>*</sup>. <a href="https://doi.org/10.12688/f1000research.26669.2">TreeSummarizedExperiment: a S4 class for data with hierarchical structure</a>. <strong><em>F1000Research</em></strong>. 2021, 9:1246.</li>
<li>S Liu, <strong>G Yu</strong><sup>*</sup>, L Liu<sup>*</sup>, X Zou, L Zhou, E Hu, Y Song. <a href="https://www.frontiersin.org/articles/10.3389/fgene.2021.625236">Identification of prognostic Stromal-Immune Score-Based Genes in Hepatocellular Carcinoma microenvironment</a>. <strong><em>Frontiers in Genetics</em></strong>. 2021, 12:625236.</li>
</ul>
<h2 id="i-classfa-fa-calendar-fa-1x-aria-hiddentruei-2020"><i class="fa fa-calendar fa-1x" aria-hidden="true"></i> 2020</h2>
<ul>
<li>J He<sup>#</sup>, S Xu<sup>#</sup>, B Zhang, C Xiao, Z Chen, F Si, J Fu, X Lin, G Zheng<sup>*</sup>, <strong>G Yu</strong><sup>*</sup>, J Chen<sup>*</sup>. <a href="https://doi.org/10.18632/aging.103168">Gut microbiota and metabolite alterations associated with reduced bone mineral density or bone metabolic indexes in postmenopausal osteoporosis</a>. <strong><em>Aging (Albany NY)</em></strong>. 2020, 12(9):8583-8604.</li>
<li><strong>G Yu</strong><sup>*</sup>. <a href="https://doi.org/10.1002/cpbi.96">Using ggtree to Visualize Data on Tree-Like Structure</a>. <strong><em>Current Protocols in Bioinformatics</em></strong>. 2020, 69(1):e96. (<em><strong>Invited paper</strong></em>)</li>
<li>LG Wang, TTY Lam, S Xu, Z Dai, L Zhou, T Feng, P Guo, CW Dunn, BR Jones, T Bradley, H Zhu, Y Guan, Y Jiang, <strong>G Yu</strong><sup>*</sup>. <a href="https://doi.org/10.1093/molbev/msz240">treeio: an R package for phylogenetic tree input and output with richly annotated and associated data</a>. <strong><em>Molecular Biology and Evolution</em></strong>. 2020, 37(2):599-603.</li>
<li>Z Hao, D Lv, Y Ge, J Shi, D Weijers, <strong>G Yu</strong><sup>*</sup>, J Chen<sup>*</sup>. <a href="https://peerj.com/articles/cs-251/">RIdeogram: drawing SVG graphics to visualize and map genome-wide data on the idiograms</a>. <strong><em>PeerJ Computer Science</em></strong>. 2020, 6:e251.   <i class="fa fa-trophy fa-2x" aria-hidden="true" style="color:#CC5500;"></i> <font style="color:#4682b4;"><strong>Highly Cited Paper</strong></font></li>
</ul>
<h2 id="i-classfa-fa-calendar-fa-1x-aria-hiddentruei-2018"><i class="fa fa-calendar fa-1x" aria-hidden="true"></i> 2018</h2>
<ul>
<li><strong>G Yu</strong><sup>*</sup>, TTY Lam, H Zhu, Y Guan<sup>*</sup>. <a href="https://academic.oup.com/mbe/article-abstract/35/12/3041/5142656">Two methods for mapping and visualizing associated data on phylogeny using ggtree</a>. <strong><em>Molecular Biology and Evolution</em></strong>. 2018, 35(12):3041-3043.</li>
<li><strong>G Yu</strong>. <a href="https://doi.org/10.1093/bioinformatics/bty410">Using meshes for MeSH term enrichment and semantic analyses</a>. <strong><em>Bioinformatics</em></strong>. 2018, 34(21):3766-3767.</li>
</ul>
<h2 id="i-classfa-fa-bar-chart-fa-1x-aria-hiddentruei-selected-publications-2010-2017"><i class="fa fa-bar-chart fa-1x" aria-hidden="true"></i> Selected publications (2010-2017)</h2>
<ul>
<li><strong>G Yu</strong>, DK Smith, H Zhu, Y Guan, TTY Lam<sup>*</sup>. <a href="http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/abstract">ggtree: an R package for visualization and annotation of phylogenetic trees with their covariates and other associated data</a>. <strong><em>Methods in Ecology and Evolution</em></strong>. 2017, 8(1):28-36. (<strong><em>Feature article</em></strong> <sup id="fnref:2"><a href="#fn:2" class="footnote-ref" role="doc-noteref">2</a></sup>)  <i class="fa fa-trophy fa-2x" aria-hidden="true" style="color:#CC5500;"></i> <font style="color:#4682b4;"><strong>Highly Cited Paper</strong></font></li>
<li><strong>G Yu</strong>, QY He<sup>*</sup>. <a href="http://pubs.rsc.org/en/Content/ArticleLanding/2016/MB/C5MB00663E">ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization</a>. <strong><em>Molecular BioSystems</em></strong>. 2016, 12(2):477-479.  <i class="fa fa-trophy fa-2x" aria-hidden="true" style="color:#CC5500;"></i> <font style="color:#4682b4;"><strong>Highly Cited Paper</strong></font></li>
<li><strong>G Yu</strong><sup>*</sup>, LG Wang, QY He<sup>*</sup>. <a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btv145">ChIPseeker: an R/Bioconductor package for ChIP peak annotation, comparision and visualization</a>. <strong><em>Bioinformatics</em></strong>. 2015, 31(14):2382-2383.   <i class="fa fa-trophy fa-2x" aria-hidden="true" style="color:#CC5500;"></i> <font style="color:#4682b4;"><strong>Highly Cited Paper</strong></font></li>
<li><strong>G Yu</strong><sup>*</sup>, LG Wang, GR Yan, QY He<sup>*</sup>. <a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btu684">DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis</a>. <strong><em>Bioinformatics</em></strong>. 2015, 31(4):608-609.   <i class="fa fa-trophy fa-2x" aria-hidden="true" style="color:#CC5500;"></i> <font style="color:#4682b4;"><strong>Highly Cited Paper</strong></font></li>
<li><strong>G Yu</strong>, LG Wang, Y Han, QY He<sup>*</sup>. <a href="http://online.liebertpub.com/doi/abs/10.1089/omi.2011.0118">clusterProfiler: an R package for comparing biological themes among gene clusters</a>. <strong><em>OMICS: A Journal of Integrative Biology</em></strong>. 2012, 16(5):284-287.   <i class="fa fa-trophy fa-2x" aria-hidden="true" style="color:#CC5500;"></i> <font style="color:#4682b4;"><strong>Highly Cited Paper</strong></font></li>
<li><strong>G Yu</strong><sup>#</sup>, F Li<sup>#</sup>, Y Qin, X Bo<sup>*</sup>, Y Wu, S Wang<sup>*</sup>. <a href="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/26/7/976">GOSemSim: an R package for measuring semantic similarity among GO terms and gene products</a>. <strong><em>Bioinformatics</em></strong>. 2010, 26(7):976-978.   <i class="fa fa-trophy fa-2x" aria-hidden="true" style="color:#CC5500;"></i> <font style="color:#4682b4;"><strong>Highly Cited Paper</strong></font></li>
</ul>
<section class="footnotes" role="doc-endnotes">
<hr>
<ol>
<li id="fn:1" role="doc-endnote">
<p><a href="https://doi.org/10.1016/j.xinn.2021.100141">clusterProfiler (Wu et al. 2021)</a>, the featured article in <a href="/images/cell+press-screenshot-2021-08-24.jpg">2021-08-24âs website banner of Cell Press</a>.&#160;<a href="#fnref:1" class="footnote-backref" role="doc-backlink">&#x21a9;&#xfe0e;</a></p>
</li>
<li id="fn:2" role="doc-endnote">
<p>The <a href="http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/abstract">ggtree (Yu <em>et al.</em> 2017)</a> paper was selected as a feature article to celebrate the 10<sup>th</sup> anniversary of the launch of <strong><em><a href="https://methodsblog.com/2020/11/19/ggtree-tree-visualization/">Methods in Ecology and Evolution</a></em></strong>.&#160;<a href="#fnref:2" class="footnote-backref" role="doc-backlink">&#x21a9;&#xfe0e;</a></p>
</li>
</ol>
</section>

  </article>
</section>

  

      </div>

      
  <footer class="footer">
    <section class="container">
      
        <p>You are welcome to join us.</p>
      
      
        Â©
        
          2018 -
        
        2022
        â¥ï¸
         YuLab-SMU 
      
 
      
 
      
      
    </section>
  </footer>


    </main>

    
      
        
        <script src="/js/dark-mode.min.1d139a81392167c6a37135df59f9fd95747abba988ba33c1a557c573776cb197.js"></script>
      
    

    

    

    

    

    <script data-goatcounter="https://code.goatcounter.com/count"
        async src="//gc.zgo.at/count.js"></script>


    
<script defer src='https://static.cloudflareinsights.com/beacon.min.js'
        data-cf-beacon='{"token": "token"}'></script>


  </body>

</html>
